{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Peak and Motif detection with GEM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "GEM command line options (see more options at our website)\n",
      "   Required parameters:\n",
      "      --d <read spatial distribution file>\n",
      "      --exptX <aligned read file for expt (X is condition name)>\n",
      "   Required GEM motif discovery parameters, optional for GPS-only analysis:\n",
      "      --k <length of the k-mer for motif finding, use --k or (--kmin & --kmax)>\n",
      "      --k_min <min value of k, e.g. 6>\n",
      "      --k_max <max value of k, e.g. 13>\n",
      "      --seed <exact k-mer string to jump start k-mer set motif discovery>\n",
      "      --genome <the path to the genome sequence directory, for motif finding>\n",
      "   Optional parameters:\n",
      "      --ctrlX <aligned reads file for ctrl (for each condition, ctrlX should match exptX)>\n",
      "      --g <genome chrom.sizes file with chr name/length pairs>\n",
      "      --f <read file format, BED/SAM/BOWTIE/ELAND/NOVO (default BED)>\n",
      "      --s <size of mappable genome in bp (default is estimated from genome chrom sizes)>\n",
      "      --a <minimum alpha value for sparse prior (default is esitmated from the whole dataset coverage)>\n",
      "      --q <significance level for q-value, specify as -log10(q-value) (default=2, q-value=0.01)>\n",
      "      --t <maximum number of threads to run GEM in paralell (default=#CPU)>\n",
      "      --out <output folder name and file name prefix>\n",
      "      --k_seqs <number of binding events to use for motif discovery (default=5000)>\n",
      "   Optional flags: \n",
      "      --fa use a fixed user-specified alpha value for all the regions\n",
      "      --help print this help information and exit\n",
      "\n",
      "Example: java -Xmx10G -jar gem.jar --d Read_Distribution_default.txt --g mm8.chrom.sizes --genome your_path/mm8 --s 2000000000 --expt SRX000540_mES_CTCF.bed --ctrl SRX000543_mES_GFP.bed --f BED --out mouseCTCF --k_min 6 --k_max 13\n"
     ]
    }
   ],
   "source": [
    "# run GEM without any parameters\n",
    "!java -jar /usr/local/gem/gem.jar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/06 18:29:15\n",
      "\n",
      "Warning: no options (--k, --k_min & --k_max, or --seed) to run motif discovery. GEM will run GPS and stop!\n",
      "The read distribution file is required. Use --d option.\n",
      "\n",
      "GEM command line options (see more options at our website)\n",
      "   Required parameters:\n",
      "      --d <read spatial distribution file>\n",
      "      --exptX <aligned read file for expt (X is condition name)>\n",
      "   Required GEM motif discovery parameters, optional for GPS-only analysis:\n",
      "      --k <length of the k-mer for motif finding, use --k or (--kmin & --kmax)>\n",
      "      --k_min <min value of k, e.g. 6>\n",
      "      --k_max <max value of k, e.g. 13>\n",
      "      --seed <exact k-mer string to jump start k-mer set motif discovery>\n",
      "      --genome <the path to the genome sequence directory, for motif finding>\n",
      "   Optional parameters:\n",
      "      --ctrlX <aligned reads file for ctrl (for each condition, ctrlX should match exptX)>\n",
      "      --g <genome chrom.sizes file with chr name/length pairs>\n",
      "      --f <read file format, BED/SAM/BOWTIE/ELAND/NOVO (default BED)>\n",
      "      --s <size of mappable genome in bp (default is estimated from genome chrom sizes)>\n",
      "      --a <minimum alpha value for sparse prior (default is esitmated from the whole dataset coverage)>\n",
      "      --q <significance level for q-value, specify as -log10(q-value) (default=2, q-value=0.01)>\n",
      "      --t <maximum number of threads to run GEM in paralell (default=#CPU)>\n",
      "      --out <output folder name and file name prefix>\n",
      "      --k_seqs <number of binding events to use for motif discovery (default=5000)>\n",
      "   Optional flags: \n",
      "      --fa use a fixed user-specified alpha value for all the regions\n",
      "      --help print this help information and exit\n",
      "\n",
      "Example: java -Xmx10G -jar gem.jar --d Read_Distribution_default.txt --g mm8.chrom.sizes --genome your_path/mm8 --s 2000000000 --expt SRX000540_mES_CTCF.bed --ctrl SRX000543_mES_GFP.bed --f BED --out mouseCTCF --k_min 6 --k_max 13\n",
      "\n",
      "Your input options are:\n",
      "--t 1 \n",
      "\n"
     ]
    }
   ],
   "source": [
    "# add parameters that restrict memor and cpu usage\n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar --t 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/06 18:31:54\n",
      "\n",
      "Warning: no options (--k, --k_min & --k_max, or --seed) to run motif discovery. GEM will run GPS and stop!\n",
      "Estimating chromosome sizes from all read files (skip this step by adding a --g option)...\n",
      "\n",
      "Loading data...\n",
      "    Loading reads from: chipA.bam ... Loaded\n",
      "\n",
      "Options:\n",
      "--t 1 --expt chipA.bam --f BAM --d /usr/local/gem/Read_Distribution_default.txt \n",
      "\n",
      "Mappable Genome Length is 9,881,843.\n",
      "\n",
      "Getting 5' positions of all reads...\n",
      "\n",
      "Original read count stats:\n",
      "_IP  \tBases: 67177\tHitCounts: 82072.0\n",
      "\n",
      "Filter duplicate reads.\n",
      "_IP  \tBases: 67177\tHitCounts: 79786.0\n",
      "\n",
      "At Poisson p-value 1.0e-03, in a 501bp window, expect 12 reads.\n",
      "\n",
      "\n",
      "Sorting reads and selecting enriched regions ...\n",
      "\n",
      "The genome is segmented into 471 regions for analysis.\n",
      "\n",
      "============================ Round 0 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/471\t1.1m\n",
      "200\t/471\t1.1m\n",
      "300\t/471\t1.2m\n",
      "471\t/471\t1.2m\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t81\n",
      "Insignificant:\t215\n",
      "Filtered:\t6\n",
      "\n",
      "Total 31 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 1.2m\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (81<500) significant events.\n",
      "\n",
      "============================ Round 1 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/471\t16.0s\n",
      "200\t/471\t20.0s\n",
      "300\t/471\t24.0s\n",
      "400\t/471\t27.0s\n",
      "471\t/471\t32.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t81\n",
      "Insignificant:\t215\n",
      "Filtered:\t6\n",
      "\n",
      "Total 31 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 32.0s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (81<500) significant events.\n",
      "\n",
      "Finished! GPS analysis results are printed to:\n",
      "/home/boycem4/chip_seq_project/DAY 4/out/out.GPS_events.txt\n",
      "/home/boycem4/chip_seq_project/DAY 4/out/out_outputs (folder with all other files)\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "End time: 2022/01/06 18:33:36\n",
      "\n",
      "Total running time: 1.7m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# add miminum set of parameters for a peak detection run\n",
    "    #use 3gb ram max = Xmx4g\n",
    "    #use 1 cpu only = --t 1\n",
    "    #data from chromosome A (chipA.bam) only\n",
    "    #output file = bam format\n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar \\\n",
    "    --t 1 \\\n",
    "    --expt chipA.bam --f BAM \\\n",
    "    --d /usr/local/gem/Read_Distribution_default.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 572K\n",
      "-rw-r--r-- 1 boycem4 GEU3302521 7.5K Dec  3 11:06 out.GEM_events.txt\n",
      "-rw-r--r-- 1 boycem4 GEU3302521 7.1K Jan  6 18:33 out.GPS_events.txt\n",
      "drwxr-xr-x 3 boycem4 GEU3302521 4.0K Dec  3 11:06 out_outputs\n",
      "-rw-r--r-- 1 boycem4 GEU3302521  301 Dec  3 11:06 out.results.htm\n",
      "-rw-r--r-- 1 boycem4 GEU3302521 547K Dec  3 11:11 out.results.html\n"
     ]
    }
   ],
   "source": [
    "# list content of newly created output directory\n",
    "!ls -lh out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Position\tIP\tControl\tFold\tExpectd\tQ_-lg10\tP_-lg10\tP_poiss\tIPvsEMP\tNoise\tKmerGroup\tMotifId\tKG_score\tStrand\n",
      "A:162945\t 3536.7\tNaN\tNaN\t   81.2\t 999.00\tNaN\t 999.00\t  -0.73\t   0.01\t--------\t-1\t0.00\t*\n",
      "A:163049\t  881.9\tNaN\tNaN\t   81.2\t 999.00\tNaN\t 999.00\t  -0.46\t   0.01\t--------\t-1\t0.00\t*\n",
      "A:162840\t  741.4\tNaN\tNaN\t   81.2\t 999.00\tNaN\t 999.00\t  -0.49\t   0.01\t--------\t-1\t0.00\t*\n",
      "A:3223\t  674.2\tNaN\tNaN\t   81.2\t 999.00\tNaN\t 999.00\t  -0.71\t   0.18\t--------\t-1\t0.00\t*\n",
      "A:2840\t  531.0\tNaN\tNaN\t   81.2\t 238.52\tNaN\t 240.30\t  -0.76\t   0.18\t--------\t-1\t0.00\t*\n",
      "A:3056\t  484.9\tNaN\tNaN\t   81.2\t 201.14\tNaN\t 202.84\t  -0.80\t   0.18\t--------\t-1\t0.00\t*\n",
      "A:2665\t  267.7\tNaN\tNaN\t   81.2\t  57.84\tNaN\t  59.47\t  -0.61\t   0.18\t--------\t-1\t0.00\t*\n",
      "A:3487\t  257.6\tNaN\tNaN\t   81.2\t  52.79\tNaN\t  54.36\t  -0.72\t   0.18\t--------\t-1\t0.00\t*\n",
      "A:334019\t  228.0\tNaN\tNaN\t   81.2\t  38.54\tNaN\t  40.06\t  -0.36\t   0.81\t--------\t-1\t0.00\t*\n"
     ]
    }
   ],
   "source": [
    "# bring up first 10 lines of output file\n",
    "!head out/out.GPS_events.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/11 18:50:37\n",
      "\n",
      "Loading data...\n",
      "    Loading reads from: chipA.bam ... WARNING: BAM index file /home/boycem4/chip_seq_project/all/chipA.bam.bai is older than BAM /home/boycem4/chip_seq_project/all/chipA.bam\n",
      "Loaded\n",
      "\n",
      "Options:\n",
      "--t 1 --expt chipA.bam --out out1 --f BAM --d /usr/local/gem/Read_Distribution_default.txt --genome . --k 10 --g chrom_sizes.txt \n",
      "\n",
      "Mappable Genome Length is 9,870,643.\n",
      "\n",
      "Getting 5' positions of all reads...\n",
      "\n",
      "Original read count stats:\n",
      "_IP  \tBases: 67111\tHitCounts: 82038.0\n",
      "\n",
      "Filter duplicate reads.\n",
      "_IP  \tBases: 67111\tHitCounts: 79757.0\n",
      "\n",
      "At Poisson p-value 1.0e-03, in a 501bp window, expect 12 reads.\n",
      "\n",
      "\n",
      "Sorting reads and selecting enriched regions ...\n",
      "\n",
      "The genome is segmented into 472 regions for analysis.\n",
      "\n",
      "============================ Round 0 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/472\t1.1m\n",
      "200\t/472\t1.2m\n",
      "472\t/472\t1.2m\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t83\n",
      "Insignificant:\t197\n",
      "Filtered:\t7\n",
      "\n",
      "Total 34 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 1.2m\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (83<500) significant events.\n",
      "\n",
      "============================ Round 1 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/472\t5.0s\n",
      "200\t/472\t8.0s\n",
      "300\t/472\t11.0s\n",
      "400\t/472\t14.0s\n",
      "472\t/472\t20.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t83\n",
      "Insignificant:\t197\n",
      "Filtered:\t7\n",
      "\n",
      "Total 34 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 20.0s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (83<500) significant events.\n",
      "Loading genome sequences ...\n",
      "Done, 40ms\n",
      "GC content=0.41\n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.52. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 280+/280- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=99\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=1, base kmer=2\n",
      "k=10+4, gapped kmer=2, base kmer=4\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 99\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "AGTGATGCCG/CGGCATCACT\t4\t5\t0\t-2.4    \t6\t8.4\t8.0\t66.9\t7\n",
      "AGCCAAGTGA/TCACTTGGCT\t4\t5\t0\t-2.4    \t5\t7.1\t6.0\t42.4\t4\n",
      "CGAGGCCGAC/GTCGGCCTCG\t3\t4\t0\t-2.1    \t34\t7.2\t5.0\t36.1\t8\n",
      "CTTCAATGGG/CCCATTGAAG\t3\t4\t0\t-2.1    \t38\t6.6\t5.0\t33.2\t4\n",
      "CAGAAGCGAA/TTCGCTTCTG\t3\t3\t0\t-1.8    \t70\t5.7\t5.0\t28.7\t4\n",
      "AAGTGATGCC/GGCATCACTT\t5\t6\t0\t-2.7\n",
      "\n",
      "Align and cluster 35 k-mers with seed AGTGATGCCG,   \t#0\n",
      "Align and cluster 23 k-mers with seed AGCCAAGTGA,   \t#1\n",
      "Align and cluster 37 k-mers with seed CGAGGCCGAC,   \t#2\n",
      "Align and cluster 21 k-mers with seed CTTCAATGGG,   \t#3\n",
      "Align and cluster 19 k-mers with seed CAGAAGCGAA,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= CAGAAGCGAA \t1.31, 9, kAUC=7.7\tPWM= CAGCAGCGGA\t10.25, 35, pAUC=11.1\n",
      "#1\tk=10\tKSM= CTTCAATGGG \t1.00, 7, kAUC=7.0\tPWM= GCTCAACGGG\t9.21, 19, pAUC=6.5\n",
      "#2\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.7\tPWM= AGTGATGCCG\t10.97, 19, pAUC=9.4\n",
      "#3\tk=10\tKSM= AGCCAAGTGA \t1.00, 6, kAUC=6.7\tPWM= AGCCAAGTGG\t11.48, 25, pAUC=9.4\n",
      "#4\tk=10\tKSM= CGAGGCCGAC \t1.00, 6, kAUC=6.7\tPWM= CGAGGCCGAC\t11.31, 16, pAUC=7.0\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= CAGAAGCGAA \t1.31, 9, kAUC=7.7\tPWM= CAGCAGCGGA\t10.25, 35, pAUC=11.1\n",
      "#1\tk=10\tKSM= CTTCAATGGG \t1.00, 7, kAUC=7.0\tPWM= GCTCAACGGG\t9.21, 19, pAUC=6.5\n",
      "#2\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.7\tPWM= AGTGATGCCG\t10.97, 19, pAUC=9.4\n",
      "#3\tk=10\tKSM= AGCCAAGTGA \t1.00, 6, kAUC=6.7\tPWM= AGCCAAGTGG\t11.48, 25, pAUC=9.4\n",
      "#4\tk=10\tKSM= CGAGGCCGAC \t1.00, 6, kAUC=6.7\tPWM= CGAGGCCGAC\t11.31, 16, pAUC=7.0\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out1_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= CAGAAGCGAA \t1.31, 9, kAUC=7.7\tPWM= CAGCAGCGGA\t10.25, 35, pAUC=11.1\n",
      "#1\tk=10\tKSM= CTTCAATGGG \t1.00, 7, kAUC=7.0\tPWM= GCTCAACGGG\t9.21, 19, pAUC=6.5\n",
      "#2\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.7\tPWM= AGTGATGCCG\t10.97, 19, pAUC=9.4\n",
      "#3\tk=10\tKSM= AGCCAAGTGA \t1.00, 6, kAUC=6.7\tPWM= AGCCAAGTGG\t11.48, 25, pAUC=9.4\n",
      "#4\tk=10\tKSM= CGAGGCCGAC \t1.00, 6, kAUC=6.7\tPWM= CGAGGCCGAC\t11.31, 16, pAUC=7.0\n",
      "\n",
      "\n",
      "------------------------- out1_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= CAGAAGCGAA \t1.31, 9, kAUC=7.7\tPWM= CAGCAGCGGA\t10.25, 35, pAUC=11.1\n",
      "#1\tk=10\tKSM= CTTCAATGGG \t1.00, 7, kAUC=7.0\tPWM= GCTCAACGGG\t9.21, 19, pAUC=6.5\n",
      "#2\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.7\tPWM= AGTGATGCCG\t10.97, 19, pAUC=9.4\n",
      "#3\tk=10\tKSM= AGCCAAGTGA \t1.00, 6, kAUC=6.7\tPWM= AGCCAAGTGG\t11.48, 25, pAUC=9.4\n",
      "#4\tk=10\tKSM= CGAGGCCGAC \t1.00, 6, kAUC=6.7\tPWM= CGAGGCCGAC\t11.31, 16, pAUC=7.0\n",
      "\n",
      "------------------------- out1_1, k = ALL (after merging) -------------------------\n",
      "#0\tk=10\tKSM= CAGAAGCGAA \t1.31, 9, kAUC=7.7\tPWM= CAGCAGCGGA\t10.25, 35, pAUC=11.1\n",
      "#1\tk=10\tKSM= CTTCAATGGG \t1.00, 7, kAUC=7.0\tPWM= GCTCAACGGG\t9.21, 19, pAUC=6.5\n",
      "#2\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.7\tPWM= AGTGATGCCG\t10.97, 19, pAUC=9.4\n",
      "#3\tk=10\tKSM= AGCCAAGTGA \t1.00, 6, kAUC=6.7\tPWM= AGCCAAGTGG\t11.48, 25, pAUC=9.4\n",
      "#4\tk=10\tKSM= CGAGGCCGAC \t1.00, 6, kAUC=6.7\tPWM= CGAGGCCGAC\t11.31, 16, pAUC=7.0\n",
      "\n",
      "38ms: Finalizing 5 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out1_1 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out1_1 motif #0\n",
      "\n",
      "KSM top k-mer: CAGAAGCGAA/TTCGCTTCTG, total 5 k-mers\n",
      "KSM threshold: 1.31, \t\thit=9+/0-, kAUC=7.7\n",
      "\n",
      "  |\n",
      "CAGCAGCGGA\n",
      "G  A    A \n",
      "          \n",
      "          \n",
      "PWM threshold: 10.25/16.34, \thit=35+/10-, pAUC=11.1\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out1_1 motif #1\n",
      "\n",
      "KSM top k-mer: CTTCAATGGG/CCCATTGAAG, total 6 k-mers\n",
      "KSM threshold: 1.00, \t\thit=7+/0-, kAUC=7.0\n",
      "\n",
      "GCTCAACGGG\n",
      "CGC   T   \n",
      " T\n",
      "PWM threshold: 9.21/13.47, \thit=19+/15-, pAUC=6.5\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out1_1 motif #2\n",
      "\n",
      "KSM top k-mer: AGTGATGCCG/CGGCATCACT, total 11 k-mers\n",
      "KSM threshold: 1.00, \t\thit=6+/0-, kAUC=6.7\n",
      "\n",
      "              |\n",
      "AGTGATGCCG\n",
      "          \n",
      "          \n",
      "          \n",
      "PWM threshold: 10.97/17.90, \thit=19+/2-, pAUC=9.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out1_1 motif #3\n",
      "\n",
      "KSM top k-mer: AGCCAAGTGA/TCACTTGGCT, total 7 k-mers\n",
      "KSM threshold: 1.00, \t\thit=6+/0-, kAUC=6.7\n",
      "\n",
      "                   |\n",
      "AGCCAAGTGG\n",
      "G    T   A\n",
      "          \n",
      "          \n",
      "PWM threshold: 11.48/16.41, \thit=25+/1-, pAUC=9.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out1_1 motif #4\n",
      "\n",
      "KSM top k-mer: CGAGGCCGAC/GTCGGCCTCG, total 13 k-mers\n",
      "KSM threshold: 1.00, \t\thit=6+/0-, kAUC=6.7\n",
      "\n",
      "         |\n",
      "CGAGGCCGAC\n",
      "          \n",
      "          \n",
      "          \n",
      "PWM threshold: 11.31/18.36, \thit=16+/3-, pAUC=7.0\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "============================ Round 2 ============================\n",
      "Running EM with motif positional prior ...\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/472\t6.0s\n",
      "200\t/472\t16.0s\n",
      "300\t/472\t19.0s\n",
      "400\t/472\t22.0s\n",
      "472\t/472\t28.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t82\n",
      "Insignificant:\t208\n",
      "Filtered:\t5\n",
      "\n",
      "Total 32 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 28.0s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (82<500) significant events.\n",
      "Percentage of significant events with a motif match, divided in 9 bins (~9 events per bin):\n",
      "0 66 11 11 0 0 0 0 0 0 \n",
      "\n",
      "Percentage of filtered events with a motif match, divided in 5 bins (~1 events per bin):\n",
      "0 0 0 0 0 \n",
      "\n",
      "Percentage of insignificant events with a motif match, divided in 10 bins (~21 events per bin):\n",
      "4 4 0 0 4 0 0 0 0 5 \n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.52. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 290+/290- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=112\n",
      "k=10+1, gapped kmer=3, base kmer=6\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=1, base kmer=3\n",
      "k=10+4, gapped kmer=3, base kmer=6\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 112\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "CACCAGCAGC/GCTGCTGGTG\t6\t7\t0\t-3.0    \t0\t13.2\t8.0\t105.8\t15\n",
      "AGCCAAGTGA/TCACTTGGCT\t4\t6\t0\t-2.7    \t5\t7.7\t6.0\t46.5\t4\n",
      "AGTGATGCCG/CGGCATCACT\t4\t6\t0\t-2.7    \t6\t9.2\t4.0\t36.7\t7\n",
      "GTACACGCCG/CGGCGTGTAC\t3\t4\t0\t-2.1    \t56\t6.6\t5.0\t33.2\t5\n",
      "CAGTTGCCAC/GTGGCAACTG\t3\t3\t0\t-1.8    \t81\t5.7\t4.0\t23.0\t3\n",
      "\n",
      "Align and cluster 48 k-mers with seed CACCAGCAGC,   \t#0\n",
      "Align and cluster 22 k-mers with seed AGCCAAGTGA,   \t#1\n",
      "Align and cluster 38 k-mers with seed AGTGATGCCG,   \t#2\n",
      "Align and cluster 27 k-mers with seed GTACACGCCG,   \t#3\n",
      "Align and cluster 33 k-mers with seed CAGTTGCCAC,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= CACCAGCAGC \t1.10, 22, kAUC=11.6\tPWM= CACCAGCAGC\t10.16, 35, pAUC=11.5\n",
      "#1\tk=10\tKSM= CAGTTGCCAC \t1.10, 20, kAUC=10.9\tPWM= CAGTTGCCAC\t9.39, 38, pAUC=8.9\n",
      "#2\tk=10\tKSM= AGCCAAGTGA \t1.00, 10, kAUC=7.6\tPWM= AGCCAAGTGG\t9.30, 29, pAUC=8.8\n",
      "#3\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.6\tPWM= AGTGATGCCG\t9.19, 25, pAUC=10.6\n",
      "#4\tk=10\tKSM= GTACACGCCG \t1.31, 4, kAUC=6.0\tPWM= GTACACGCCG\t9.82, 14, pAUC=6.3\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= CACCAGCAGC \t1.10, 22, kAUC=11.6\tPWM= CACCAGCAGC\t10.16, 35, pAUC=11.5\n",
      "#1\tk=10\tKSM= CAGTTGCCAC \t1.10, 20, kAUC=10.9\tPWM= CAGTTGCCAC\t9.39, 38, pAUC=8.9\n",
      "#2\tk=10\tKSM= AGCCAAGTGA \t1.00, 10, kAUC=7.6\tPWM= AGCCAAGTGG\t9.30, 29, pAUC=8.8\n",
      "#3\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.6\tPWM= AGTGATGCCG\t9.19, 25, pAUC=10.6\n",
      "#4\tk=10\tKSM= GTACACGCCG \t1.31, 4, kAUC=6.0\tPWM= GTACACGCCG\t9.82, 14, pAUC=6.3\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out1_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= CACCAGCAGC \t1.10, 22, kAUC=11.6\tPWM= CACCAGCAGC\t10.16, 35, pAUC=11.5\n",
      "#1\tk=10\tKSM= CAGTTGCCAC \t1.10, 20, kAUC=10.9\tPWM= CAGTTGCCAC\t9.39, 38, pAUC=8.9\n",
      "#2\tk=10\tKSM= AGCCAAGTGA \t1.00, 10, kAUC=7.6\tPWM= AGCCAAGTGG\t9.30, 29, pAUC=8.8\n",
      "#3\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.6\tPWM= AGTGATGCCG\t9.19, 25, pAUC=10.6\n",
      "#4\tk=10\tKSM= GTACACGCCG \t1.31, 4, kAUC=6.0\tPWM= GTACACGCCG\t9.82, 14, pAUC=6.3\n",
      "\n",
      "\n",
      "------------------------- out1_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= CACCAGCAGC \t1.10, 22, kAUC=11.6\tPWM= CACCAGCAGC\t10.16, 35, pAUC=11.5\n",
      "#1\tk=10\tKSM= CAGTTGCCAC \t1.10, 20, kAUC=10.9\tPWM= CAGTTGCCAC\t9.39, 38, pAUC=8.9\n",
      "#2\tk=10\tKSM= AGCCAAGTGA \t1.00, 10, kAUC=7.6\tPWM= AGCCAAGTGG\t9.30, 29, pAUC=8.8\n",
      "#3\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.6\tPWM= AGTGATGCCG\t9.19, 25, pAUC=10.6\n",
      "#4\tk=10\tKSM= GTACACGCCG \t1.31, 4, kAUC=6.0\tPWM= GTACACGCCG\t9.82, 14, pAUC=6.3\n",
      "\n",
      "------------------------- out1_2, k = ALL (after merging) -------------------------\n",
      "#0\tk=10\tKSM= CACCAGCAGC \t1.10, 22, kAUC=11.6\tPWM= CACCAGCAGC\t10.16, 35, pAUC=11.5\n",
      "#1\tk=10\tKSM= CAGTTGCCAC \t1.10, 20, kAUC=10.9\tPWM= CAGTTGCCAC\t9.39, 38, pAUC=8.9\n",
      "#2\tk=10\tKSM= AGCCAAGTGA \t1.00, 10, kAUC=7.6\tPWM= AGCCAAGTGG\t9.30, 29, pAUC=8.8\n",
      "#3\tk=10\tKSM= AGTGATGCCG \t1.00, 6, kAUC=6.6\tPWM= AGTGATGCCG\t9.19, 25, pAUC=10.6\n",
      "#4\tk=10\tKSM= GTACACGCCG \t1.31, 4, kAUC=6.0\tPWM= GTACACGCCG\t9.82, 14, pAUC=6.3\n",
      "\n",
      "51ms: Finalizing 5 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out1_2 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out1_2 motif #0\n",
      "\n",
      "KSM top k-mer: CACCAGCAGC/GCTGCTGGTG, total 18 k-mers\n",
      "KSM threshold: 1.10, \t\thit=22+/0-, kAUC=11.6\n",
      "\n",
      "     |\n",
      "CACCAGCAGC\n",
      "  G       \n",
      "          \n",
      "          \n",
      "PWM threshold: 10.16/18.90, \thit=35+/10-, pAUC=11.5\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out1_2 motif #1\n",
      "\n",
      "KSM top k-mer: CAGTTGCCAC/GTGGCAACTG, total 11 k-mers\n",
      "KSM threshold: 1.10, \t\thit=20+/0-, kAUC=10.9\n",
      "\n",
      "  |\n",
      "CAGTTGCCAC\n",
      " C G C   G\n",
      "          \n",
      "          \n",
      "PWM threshold: 9.39/14.47, \thit=38+/21-, pAUC=8.9\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out1_2 motif #2\n",
      "\n",
      "KSM top k-mer: AGCCAAGTGA/TCACTTGGCT, total 8 k-mers\n",
      "KSM threshold: 1.00, \t\thit=10+/0-, kAUC=7.6\n",
      "\n",
      "  |\n",
      "AGCCAAGTGG\n",
      "G        A\n",
      "          \n",
      "          \n",
      "PWM threshold: 9.30/16.90, \thit=29+/14-, pAUC=8.8\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out1_2 motif #3\n",
      "\n",
      "KSM top k-mer: AGTGATGCCG/CGGCATCACT, total 11 k-mers\n",
      "KSM threshold: 1.00, \t\thit=6+/0-, kAUC=6.6\n",
      "\n",
      "              |\n",
      "AGTGATGCCG\n",
      "          \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.19/17.86, \thit=25+/6-, pAUC=10.6\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out1_2 motif #4\n",
      "\n",
      "KSM top k-mer: GTACACGCCG/CGGCGTGTAC, total 1 k-mers\n",
      "KSM threshold: 1.31, \t\thit=4+/0-, kAUC=6.0\n",
      "\n",
      "            |\n",
      "GTACACGCCG\n",
      "          \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.82/17.79, \thit=14+/9-, pAUC=6.3\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "Finished! GEM analysis results are printed to:\n",
      "/home/boycem4/chip_seq_project/all/out1/out1.GEM_events.txt\n",
      "/home/boycem4/chip_seq_project/all/out1/out1.results.htm\n",
      "/home/boycem4/chip_seq_project/all/out1/out1_outputs (folder with all other files)\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "End time: 2022/01/11 18:52:39\n",
      "\n",
      "Total running time: 2.0m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#prediction of consensus motif requires additional parameters\n",
    "\n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar \\\n",
    "    --t 1 \\\n",
    "    --expt chipA.bam \\\n",
    "    --out out1 \\\n",
    "    --f BAM \\\n",
    "    --d /usr/local/gem/Read_Distribution_default.txt \\\n",
    "    --genome . \\\n",
    "    --k 10 \\\n",
    "    --g chrom_sizes.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/11 18:52:54\n",
      "\n",
      "Loading data...\n",
      "    Loading reads from: chip.bam ... WARNING: BAM index file /home/boycem4/chip_seq_project/all/chip.bam.bai is older than BAM /home/boycem4/chip_seq_project/all/chip.bam\n",
      "Loaded\n",
      "\n",
      "Options:\n",
      "--t 1 --expt chip.bam --out out2 --f BAM --d /usr/local/gem/Read_Distribution_default.txt --genome . --k 10 --g chrom_sizes.txt \n",
      "\n",
      "Mappable Genome Length is 9,870,643.\n",
      "\n",
      "Getting 5' positions of all reads...\n",
      "\n",
      "Original read count stats:\n",
      "_IP  \tBases: 1215136\tHitCounts: 1634954.0\n",
      "\n",
      "Filter duplicate reads.\n",
      "_IP  \tBases: 1215136\tHitCounts: 1587966.0\n",
      "\n",
      "At Poisson p-value 1.0e-03, in a 501bp window, expect 110 reads.\n",
      "\n",
      "\n",
      "Sorting reads and selecting enriched regions ...\n",
      "\n",
      "The genome is segmented into 10904 regions for analysis.\n",
      "\n",
      "============================ Round 0 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t46.0s\n",
      "1000\t/10904\t1.8m\n",
      "10000\t/10904\t2.2m\n",
      "10904\t/10904\t2.2m\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t307\n",
      "Insignificant:\t65\n",
      "Filtered:\t3\n",
      "\n",
      "Total 116 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 2.2m\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (307<500) significant events.\n",
      "\n",
      "============================ Round 1 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t2.0s\n",
      "1000\t/10904\t10.0s\n",
      "10000\t/10904\t2.0m\n",
      "10904\t/10904\t2.1m\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t307\n",
      "Insignificant:\t65\n",
      "Filtered:\t3\n",
      "\n",
      "Total 116 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 2.1m\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (307<500) significant events.\n",
      "Loading genome sequences ...\n",
      "Done, 334ms\n",
      "GC content=0.41\n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.53. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 372+/372- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=511\n",
      "k=10+1, gapped kmer=37, base kmer=70\n",
      "k=10+2, gapped kmer=27, base kmer=50\n",
      "k=10+3, gapped kmer=48, base kmer=82\n",
      "k=10+4, gapped kmer=68, base kmer=110\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 511\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "GTGGTGATGT/ACATCACCAC\t13\t14\t1\t-4.6    \t0\t30.2\t8.0\t241.3\t25\n",
      "ACTCCAACGG/CCGTTGGAGT\t10\t11\t0\t-4.3    \t4\t23.2\t5.0\t116.1\t17\n",
      "GCAAGCCTAC/GTAGGCTTGC\t10\t11\t0\t-4.3    \t5\t20.2\t5.0\t100.9\t12\n",
      "CCTACCACCA/TGGTGGTAGG\t11\t12\t1\t-4.0    \t2\t24.7\t3.0\t74.2\t17\n",
      "ACGGCAAGCC/GGCTTGCCGT\t8\t9\t0\t-3.7    \t16\t17.0\t4.0\t67.9\t13\n",
      "\n",
      "Align and cluster 193 k-mers with seed GTGGTGATGT,   \t#0\n",
      "Align and cluster 143 k-mers with seed ACTCCAACGG,   \t#1\n",
      "Align and cluster 115 k-mers with seed GCAAGCCTAC,   \t#2\n",
      "Align and cluster 177 k-mers with seed CCTACCACCA,   \t#3\n",
      "Align and cluster 118 k-mers with seed ACGGCAAGCC,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "52ms: Remove motif #4.\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= GTGGTGATGT \t1.31, 49, kAUC=21.2\tPWM= GTGGTGGTGT\t10.42, 68, pAUC=14.2\n",
      "#1\tk=10\tKSM= ACTCCAACGG \t1.00, 44, kAUC=16.0\tPWM= ACTCCCACGG\t9.23, 72, pAUC=16.3\n",
      "#2\tk=10\tKSM= CCTACCACCA \t1.00, 31, kAUC=11.4\tPWM= CCTACCACCA\t10.33, 44, pAUC=10.1\n",
      "#3\tk=10\tKSM= ACGGCAAGCC \t1.00, 27, kAUC=11.1\tPWM= ACGGCAAGCC\t13.21, 26, pAUC=7.9\n",
      "#4\tk=10\tKSM= GCAAGCCTAC \t1.00, 28, kAUC=10.9\tPWM= GCAAGCCTAC\t10.77, 35, pAUC=10.3\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= GTGGTGATGT \t1.31, 49, kAUC=21.2\tPWM= GTGGTGGTGT\t10.42, 68, pAUC=14.2\n",
      "#1\tk=10\tKSM= ACTCCAACGG \t1.00, 44, kAUC=16.0\tPWM= ACTCCCACGG\t9.23, 72, pAUC=16.3\n",
      "#2\tk=10\tKSM= CCTACCACCA \t1.00, 31, kAUC=11.4\tPWM= CCTACCACCA\t10.33, 44, pAUC=10.1\n",
      "#3\tk=10\tKSM= ACGGCAAGCC \t1.00, 27, kAUC=11.1\tPWM= ACGGCAAGCC\t13.21, 26, pAUC=7.9\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out2_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= GTGGTGATGT \t1.31, 49, kAUC=21.2\tPWM= GTGGTGGTGT\t10.42, 68, pAUC=14.2\n",
      "#1\tk=10\tKSM= ACTCCAACGG \t1.00, 44, kAUC=16.0\tPWM= ACTCCCACGG\t9.23, 72, pAUC=16.3\n",
      "#2\tk=10\tKSM= CCTACCACCA \t1.00, 31, kAUC=11.4\tPWM= CCTACCACCA\t10.33, 44, pAUC=10.1\n",
      "#3\tk=10\tKSM= ACGGCAAGCC \t1.00, 27, kAUC=11.1\tPWM= ACGGCAAGCC\t13.21, 26, pAUC=7.9\n",
      "\n",
      "\n",
      "------------------------- out2_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= GTGGTGATGT \t1.31, 49, kAUC=21.2\tPWM= GTGGTGGTGT\t10.42, 68, pAUC=14.2\n",
      "#1\tk=10\tKSM= ACTCCAACGG \t1.00, 44, kAUC=16.0\tPWM= ACTCCCACGG\t9.23, 72, pAUC=16.3\n",
      "#2\tk=10\tKSM= CCTACCACCA \t1.00, 31, kAUC=11.4\tPWM= CCTACCACCA\t10.33, 44, pAUC=10.1\n",
      "#3\tk=10\tKSM= ACGGCAAGCC \t1.00, 27, kAUC=11.1\tPWM= ACGGCAAGCC\t13.21, 26, pAUC=7.9\n",
      "\n",
      "------------------------- out2_1, k = ALL (after merging) -------------------------\n",
      "#0\tk=10\tKSM= GTGGTGATGT \t1.31, 49, kAUC=21.2\tPWM= GTGGTGGTGT\t10.42, 68, pAUC=14.2\n",
      "#1\tk=10\tKSM= ACTCCAACGG \t1.00, 44, kAUC=16.0\tPWM= ACTCCCACGG\t9.23, 72, pAUC=16.3\n",
      "#2\tk=10\tKSM= CCTACCACCA \t1.00, 31, kAUC=11.4\tPWM= CCTACCACCA\t10.33, 44, pAUC=10.1\n",
      "#3\tk=10\tKSM= ACGGCAAGCC \t1.00, 27, kAUC=11.1\tPWM= ACGGCAAGCC\t13.21, 26, pAUC=7.9\n",
      "\n",
      "28ms: Finalizing 4 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out2_1 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out2_1 motif #0\n",
      "\n",
      "KSM top k-mer: GTGGTGATGT/ACATCACCAC, total 107 k-mers\n",
      "KSM threshold: 1.31, \t\thit=49+/0-, kAUC=21.2\n",
      "\n",
      "  |\n",
      "GTGGTGGTGT\n",
      "      A   \n",
      "      C   \n",
      "          \n",
      "PWM threshold: 10.42/15.90, \thit=68+/22-, pAUC=14.2\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out2_1 motif #1\n",
      "\n",
      "KSM top k-mer: ACTCCAACGG/CCGTTGGAGT, total 34 k-mers\n",
      "KSM threshold: 1.00, \t\thit=44+/0-, kAUC=16.0\n",
      "\n",
      "     |\n",
      "ACTCCCACGG\n",
      "     G    \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.23/16.34, \thit=72+/20-, pAUC=16.3\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out2_1 motif #2\n",
      "\n",
      "KSM top k-mer: CCTACCACCA/TGGTGGTAGG, total 25 k-mers\n",
      "KSM threshold: 1.00, \t\thit=31+/0-, kAUC=11.4\n",
      "\n",
      "        |\n",
      "CCTACCACCA\n",
      "          \n",
      "          \n",
      "          \n",
      "PWM threshold: 10.33/18.40, \thit=44+/17-, pAUC=10.1\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out2_1 motif #3\n",
      "\n",
      "KSM top k-mer: ACGGCAAGCC/GGCTTGCCGT, total 30 k-mers\n",
      "KSM threshold: 1.00, \t\thit=27+/0-, kAUC=11.1\n",
      "\n",
      "                |\n",
      "ACGGCAAGCC\n",
      "    T  C  \n",
      "          \n",
      "          \n",
      "PWM threshold: 13.21/19.22, \thit=26+/0-, pAUC=7.9\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "============================ Round 2 ============================\n",
      "Running EM with motif positional prior ...\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t12.0s\n",
      "1000\t/10904\t27.0s\n",
      "10000\t/10904\t2.4m\n",
      "10904\t/10904\t2.5m\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t473\n",
      "Insignificant:\t103\n",
      "Filtered:\t3\n",
      "\n",
      "Total 178 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 2.5m\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (473<500) significant events.\n",
      "Percentage of significant events with a motif match, divided in 10 bins (~48 events per bin):\n",
      "35 68 66 56 47 52 33 43 31 36 \n",
      "\n",
      "Percentage of filtered events with a motif match, divided in 3 bins (~1 events per bin):\n",
      "0 0 0 \n",
      "\n",
      "Percentage of insignificant events with a motif match, divided in 9 bins (~11 events per bin):\n",
      "45 36 45 27 27 72 27 18 36 0 \n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.53. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 576+/576- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=844\n",
      "k=10+1, gapped kmer=221, base kmer=335\n",
      "k=10+2, gapped kmer=208, base kmer=305\n",
      "k=10+3, gapped kmer=226, base kmer=295\n",
      "k=10+4, gapped kmer=245, base kmer=325\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 844\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TGGTGGTGAT/ATCACCACCA\t57\t63\t0\t-20.7    \t0\t122.7\t8.0\t981.7\t101\n",
      "GCAAGCCTAC/GTAGGCTTGC\t43\t48\t0\t-15.9    \t4\t64.6\t5.0\t323.1\t15\n",
      "ACGGCAAGCC/GGCTTGCCGT\t40\t44\t0\t-14.6    \t6\t63.3\t5.0\t316.4\t17\n",
      "GTGGTAGGCT/AGCCTACCAC\t32\t35\t0\t-11.7    \t12\t58.0\t4.0\t231.9\t21\n",
      "CCATTGAAGC/GCTTCAATGG\t24\t27\t0\t-9.2    \t26\t45.3\t5.0\t226.5\t14\n",
      "\n",
      "Align and cluster 668 k-mers with seed TGGTGGTGAT,   \t#0\n",
      "Align and cluster 175 k-mers with seed GCAAGCCTAC,   \t#1\n",
      "Align and cluster 220 k-mers with seed ACGGCAAGCC,   \t#2\n",
      "Align and cluster 263 k-mers with seed GTGGTAGGCT,   \t#3\n",
      "Align and cluster 178 k-mers with seed CCATTGAAGC,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "114ms: Remove motif #1.\n",
      "189ms: Remove motif #4.\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TGGTGGTGAT \t1.17, 218, kAUC=46.6\tPWM= TGGTGGTGAT\t9.86, 220, pAUC=37.1\n",
      "#1\tk=10\tKSM= GTGGTAGGCT \t1.10, 209, kAUC=39.9\tPWM= GTGGTGGTGA\t10.53, 211, pAUC=30.2\n",
      "#2\tk=10\tKSM= CCATTGAAGC \t1.20, 94, kAUC=22.3\tPWM= CCACCGAAGC\t10.39, 96, pAUC=15.5\n",
      "#3\tk=10\tKSM= ACGGCAAGCC \t1.00, 87, kAUC=18.2\tPWM= ACGGCAAGCC\t9.44, 97, pAUC=17.2\n",
      "#4\tk=10\tKSM= GCAAGCCTAC \t1.00, 79, kAUC=16.9\tPWM= GCAAGCCTAC\t13.37, 72, pAUC=15.1\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TGGTGGTGAT \t1.17, 218, kAUC=46.6\tPWM= TGGTGGTGAT\t9.86, 220, pAUC=37.1\n",
      "#2\tk=10\tKSM= CCATTGAAGC \t1.20, 94, kAUC=22.3\tPWM= CCACCGAAGC\t10.39, 96, pAUC=15.5\n",
      "#3\tk=10\tKSM= ACGGCAAGCC \t1.00, 87, kAUC=18.2\tPWM= ACGGCAAGCC\t9.44, 97, pAUC=17.2\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out2_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TGGTGGTGAT \t1.17, 218, kAUC=46.6\tPWM= TGGTGGTGAT\t9.86, 220, pAUC=37.1\n",
      "#1\tk=10\tKSM= CCATTGAAGC \t1.20, 94, kAUC=22.3\tPWM= CCACCGAAGC\t10.39, 96, pAUC=15.5\n",
      "#2\tk=10\tKSM= ACGGCAAGCC \t1.00, 87, kAUC=18.2\tPWM= ACGGCAAGCC\t9.44, 97, pAUC=17.2\n",
      "\n",
      "\n",
      "------------------------- out2_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TGGTGGTGAT \t1.17, 218, kAUC=46.6\tPWM= TGGTGGTGAT\t9.86, 220, pAUC=37.1\n",
      "#1\tk=10\tKSM= CCATTGAAGC \t1.20, 94, kAUC=22.3\tPWM= CCACCGAAGC\t10.39, 96, pAUC=15.5\n",
      "#2\tk=10\tKSM= ACGGCAAGCC \t1.00, 87, kAUC=18.2\tPWM= ACGGCAAGCC\t9.44, 97, pAUC=17.2\n",
      "\n",
      "------------------------- out2_2, k = ALL (after merging) -------------------------\n",
      "#0\tk=10\tKSM= TGGTGGTGAT \t1.17, 218, kAUC=46.6\tPWM= TGGTGGTGAT\t9.86, 220, pAUC=37.1\n",
      "#1\tk=10\tKSM= CCATTGAAGC \t1.20, 94, kAUC=22.3\tPWM= CCACCGAAGC\t10.39, 96, pAUC=15.5\n",
      "#2\tk=10\tKSM= ACGGCAAGCC \t1.00, 87, kAUC=18.2\tPWM= ACGGCAAGCC\t9.44, 97, pAUC=17.2\n",
      "\n",
      "44ms: Finalizing 3 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out2_2 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out2_2 motif #0\n",
      "\n",
      "KSM top k-mer: TGGTGGTGAT/ATCACCACCA, total 472 k-mers\n",
      "KSM threshold: 1.17, \t\thit=218+/1-, kAUC=46.6\n",
      "\n",
      "    |\n",
      "TGGTGGTGAT\n",
      "          \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.86/18.37, \thit=220+/16-, pAUC=37.1\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out2_2 motif #1\n",
      "\n",
      "KSM top k-mer: CCATTGAAGC/GCTTCAATGG, total 59 k-mers\n",
      "KSM threshold: 1.20, \t\thit=94+/0-, kAUC=22.3\n",
      "\n",
      "|\n",
      "CCACCGAAGC\n",
      "    T GC  \n",
      "          \n",
      "          \n",
      "PWM threshold: 10.39/14.87, \thit=96+/19-, pAUC=15.5\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out2_2 motif #2\n",
      "\n",
      "KSM top k-mer: ACGGCAAGCC/GGCTTGCCGT, total 74 k-mers\n",
      "KSM threshold: 1.00, \t\thit=87+/0-, kAUC=18.2\n",
      "\n",
      "ACGGCAAGCC\n",
      "       C\n",
      "PWM threshold: 9.44/18.40, \thit=97+/18-, pAUC=17.2\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "Finished! GEM analysis results are printed to:\n",
      "/home/boycem4/chip_seq_project/all/out2/out2.GEM_events.txt\n",
      "/home/boycem4/chip_seq_project/all/out2/out2.results.htm\n",
      "/home/boycem4/chip_seq_project/all/out2/out2_outputs (folder with all other files)\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "End time: 2022/01/11 18:59:54\n",
      "\n",
      "Total running time: 7.0m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#gem run on entire genome\n",
    "\n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar \\\n",
    "    --t 1 \\\n",
    "    --expt chip.bam \\\n",
    "    --out out2 \\\n",
    "    --f BAM \\\n",
    "    --d /usr/local/gem/Read_Distribution_default.txt \\\n",
    "    --genome . \\\n",
    "    --k 10 \\\n",
    "    --g chrom_sizes.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/11 19:01:37\n",
      "\n",
      "Loading data...\n",
      "    Loading reads from: chip.bam ... WARNING: BAM index file /home/boycem4/chip_seq_project/all/chip.bam.bai is older than BAM /home/boycem4/chip_seq_project/all/chip.bam\n",
      "Loaded\n",
      "    Loading reads from: ctrl.bam ... WARNING: BAM index file /home/boycem4/chip_seq_project/all/ctrl.bam.bai is older than BAM /home/boycem4/chip_seq_project/all/ctrl.bam\n",
      "Loaded\n",
      "\n",
      "Options:\n",
      "--t 1 --expt chip.bam --ctrl ctrl.bam --out out3 --f BAM --d /usr/local/gem/Read_Distribution_default.txt --genome . --k 10 --g chrom_sizes.txt \n",
      "\n",
      "Mappable Genome Length is 9,870,643.\n",
      "\n",
      "Getting 5' positions of all reads...\n",
      "\n",
      "Original read count stats:\n",
      "_IP  \tBases: 1215136\tHitCounts: 1634954.0\n",
      "_CTRL\tBases: 1107291\tHitCounts: 1673321.0\n",
      "\n",
      "Filter duplicate reads.\n",
      "_IP  \tBases: 1215136\tHitCounts: 1587966.0\n",
      "_CTRL\tBases: 1107291\tHitCounts: 1620135.0\n",
      "\n",
      "At Poisson p-value 1.0e-03, in a 501bp window, expect 110 reads.\n",
      "\n",
      "\n",
      "\tIP: 1587966\tCtrl: 1620135\t IP/Ctrl: 0.98\n",
      "\n",
      "Sorting reads and selecting enriched regions ...\n",
      "For condition , IP/Control = 2.70\n",
      "\n",
      "The genome is segmented into 10904 regions for analysis.\n",
      "\n",
      "============================ Round 0 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t29.0s\n",
      "1000\t/10904\t33.0s\n",
      "10904\t/10904\t33.0s\n",
      "\n",
      "Scaling condition , IP/Control = 1.84\n",
      "\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t51\n",
      "Insignificant:\t15\n",
      "Filtered:\t1\n",
      "\n",
      "Total 39 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 33.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (51<500) significant events.\n",
      "\n",
      "============================ Round 1 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t1.0s\n",
      "1000\t/10904\t2.0s\n",
      "10000\t/10904\t21.0s\n",
      "10904\t/10904\t23.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t59\n",
      "Insignificant:\t20\n",
      "Filtered:\t1\n",
      "\n",
      "Total 42 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 23.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (59<500) significant events.\n",
      "Loading genome sequences ...\n",
      "Done, 353ms\n",
      "GC content=0.41\n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.52. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 79+/79- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=13\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 13\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAA/TTCCGTGGAA\t6\t8\t0\t-3.5    \t0\t11.0\t8.0\t87.6\t5\n",
      "GGTTTCCGTG/CACGGAAACC\t3\t4\t0\t-2.2    \t2\t4.9\t4.0\t19.6\t4\n",
      "ATTTAAAGGG/CCCTTTAAAT\t3\t3\t0\t-1.8    \t6\t3.0\t4.0\t12.0\t3\n",
      "\n",
      "Align and cluster 8 k-mers with seed TTCCACGGAA,   \t#0\n",
      "Align and cluster 6 k-mers with seed GGTTTCCGTG,   \t#1\n",
      "Align and cluster 6 k-mers with seed ATTTAAAGGG,   \t#2\n",
      "\n",
      "Merging redundant motifs ...\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out3_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "\n",
      "------------------------- out3_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "------------------------- out3_1, k = ALL (after merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "18ms: Finalizing 3 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out3_1 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out3_1 motif #0\n",
      "\n",
      "KSM top k-mer: TTCCACGGAA/TTCCGTGGAA, total 6 k-mers\n",
      "KSM threshold: 1.00, \t\thit=15+/0-, kAUC=20.6\n",
      "\n",
      "              |\n",
      "TTCCACGGAA\n",
      "          \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.31/17.47, \thit=32+/6-, pAUC=33.2\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out3_1 motif #1\n",
      "\n",
      "KSM top k-mer: GGTTTCCGTG/CACGGAAACC, total 4 k-mers\n",
      "KSM threshold: 1.11, \t\thit=5+/0-, kAUC=9.8\n",
      "\n",
      "GGTTTCCGTG\n",
      "C C\n",
      "PWM threshold: 10.96/17.07, \thit=18+/3-, pAUC=18.5\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out3_1 motif #2\n",
      "\n",
      "KSM top k-mer: ATTTAAAGGG/CCCTTTAAAT, total 4 k-mers\n",
      "KSM threshold: 1.23, \t\thit=3+/0-, kAUC=8.6\n",
      "\n",
      "          |\n",
      "ATTTAAAGGG\n",
      "  C  T    \n",
      "          \n",
      "          \n",
      "PWM threshold: 13.57/14.91, \thit=4+/0-, pAUC=6.0\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "============================ Round 2 ============================\n",
      "Running EM with motif positional prior ...\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t1.0s\n",
      "1000\t/10904\t2.0s\n",
      "10000\t/10904\t27.0s\n",
      "10904\t/10904\t29.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t63\n",
      "Insignificant:\t22\n",
      "Filtered:\t0\n",
      "\n",
      "Total 46 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 29.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (63<500) significant events.\n",
      "Percentage of significant events with a motif match, divided in 9 bins (~7 events per bin):\n",
      "28 42 28 42 28 14 14 14 28 \n",
      "\n",
      "Percentage of insignificant events with a motif match, divided in 7 bins (~3 events per bin):\n",
      "0 0 0 0 0 0 0 0 \n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.52. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 85+/85- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=16\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 16\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAA/TTCCGTGGAA\t8\t10\t0\t-4.1    \t0\t14.5\t8.0\t115.9\t8\n",
      "GGTGCTGCTC/GAGCAGCACC\t3\t3\t0\t-1.8    \t11\t3.0\t6.0\t18.0\t1\n",
      "GTGGGTTGTG/CACAACCCAC\t3\t3\t0\t-1.8    \t12\t3.0\t5.0\t15.0\t1\n",
      "GGTTTCCGTG/CACGGAAACC\t3\t4\t0\t-2.2    \t5\t5.3\t2.0\t10.6\t2\n",
      "CACTTCCACG/CGTGGAAGTG\t3\t3\t0\t-1.8    \t7\t3.0\t2.0\t6.0\t2\n",
      "\n",
      "Align and cluster 10 k-mers with seed TTCCACGGAA,   \t#0\n",
      "Align and cluster 1 k-mers with seed GGTGCTGCTC,   \t#1\n",
      "Align and cluster 4 k-mers with seed GTGGGTTGTG,   \t#2\n",
      "Align and cluster 6 k-mers with seed GGTTTCCGTG,   \t#3\n",
      "Align and cluster 8 k-mers with seed CACTTCCACG,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out3_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "\n",
      "------------------------- out3_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "------------------------- out3_2, k = ALL (after merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "21ms: Finalizing 5 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out3_2 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out3_2 motif #0\n",
      "\n",
      "KSM top k-mer: TTCCACGGAA/TTCCGTGGAA, total 9 k-mers\n",
      "KSM threshold: 1.00, \t\thit=19+/0-, kAUC=24.0\n",
      "\n",
      "              |\n",
      "TTCCACGGAA\n",
      "        G \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.50/17.74, \thit=33+/5-, pAUC=31.9\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out3_2 motif #1\n",
      "\n",
      "KSM top k-mer: CACTTCCACG/CGTGGAAGTG, total 8 k-mers\n",
      "KSM threshold: 1.11, \t\thit=16+/0-, kAUC=21.8\n",
      "\n",
      "                 |\n",
      "CACTTCCACG\n",
      "G GC   C  \n",
      "   A      \n",
      "          \n",
      "PWM threshold: 9.55/14.30, \thit=30+/5-, pAUC=27.5\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out3_2 motif #2\n",
      "\n",
      "KSM top k-mer: GGTTTCCGTG/CACGGAAACC, total 5 k-mers\n",
      "KSM threshold: 1.11, \t\thit=10+/0-, kAUC=15.1\n",
      "\n",
      "|\n",
      "CGTTTCCGTG\n",
      "GC        \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.34/16.51, \thit=31+/2-, pAUC=33.3\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out3_2 motif #3\n",
      "\n",
      "KSM top k-mer: GTGGGTTGTG/CACAACCCAC, total 1 k-mers\n",
      "KSM threshold: 1.23, \t\thit=3+/0-, kAUC=8.4\n",
      "\n",
      "         |\n",
      "GTGGTTTCTG\n",
      "    G  GC \n",
      "          \n",
      "          \n",
      "PWM threshold: 10.94/15.74, \thit=15+/1-, pAUC=16.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out3_2 motif #4\n",
      "\n",
      "KSM top k-mer: GGTGCTGCTC/GAGCAGCACC, total 1 k-mers\n",
      "KSM threshold: 1.34, \t\thit=3+/0-, kAUC=8.4\n",
      "\n",
      "GGTGCTGCTC\n",
      "PWM threshold: 11.01/17.77, \thit=12+/1-, pAUC=14.9\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "Finished! GEM analysis results are printed to:\n",
      "/home/boycem4/chip_seq_project/all/out3/out3.GEM_events.txt\n",
      "/home/boycem4/chip_seq_project/all/out3/out3.results.htm\n",
      "/home/boycem4/chip_seq_project/all/out3/out3_outputs (folder with all other files)\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "End time: 2022/01/11 19:03:15\n",
      "\n",
      "Total running time: 1.6m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#gem using conrtrol file and entire genome \n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar \\\n",
    "    --t 1 \\\n",
    "    --expt chip.bam \\\n",
    "    --ctrl ctrl.bam \\\n",
    "    --out out3 \\\n",
    "    --f BAM \\\n",
    "    --d /usr/local/gem/Read_Distribution_default.txt \\\n",
    "    --genome . \\\n",
    "    --k 10 \\\n",
    "    --g chrom_sizes.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/07 14:21:41\n",
      "\n",
      "Loading data...\n",
      "    Loading reads from: chipA.bam ... Loaded\n",
      "    Loading reads from: ctrlA.bam ... Loaded\n",
      "\n",
      "Options:\n",
      "--t 1 --expt chipA.bam --ctrl ctrlA.bam --out out5 --outBED --f BAM --d /usr/local/gem/Read_Distribution_default.txt --genome . --k 10 --g chrom_sizes.txt \\ \n",
      "\n",
      "Mappable Genome Length is 9,870,643.\n",
      "\n",
      "Getting 5' positions of all reads...\n",
      "\n",
      "Original read count stats:\n",
      "_IP  \tBases: 67177\tHitCounts: 82072.0\n",
      "_CTRL\tBases: 67138\tHitCounts: 82725.0\n",
      "\n",
      "Filter duplicate reads.\n",
      "_IP  \tBases: 67177\tHitCounts: 79786.0\n",
      "_CTRL\tBases: 67138\tHitCounts: 80370.0\n",
      "\n",
      "At Poisson p-value 1.0e-03, in a 501bp window, expect 12 reads.\n",
      "\n",
      "\n",
      "\tIP: 79786\tCtrl: 80370\t IP/Ctrl: 0.99\n",
      "\n",
      "Sorting reads and selecting enriched regions ...\n",
      "For condition , IP/Control = 1.00\n",
      "\n",
      "The genome is segmented into 471 regions for analysis.\n",
      "\n",
      "============================ Round 0 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/471\t10.0s\n",
      "200\t/471\t14.0s\n",
      "300\t/471\t15.0s\n",
      "471\t/471\t15.0s\n",
      "\n",
      "Scaling condition , IP/Control = 1.00\n",
      "\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t4\n",
      "Insignificant:\t47\n",
      "Filtered:\t0\n",
      "\n",
      "Total 2 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 15.0s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (4<500) significant events.\n",
      "\n",
      "============================ Round 1 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/471\t2.0s\n",
      "200\t/471\t14.0s\n",
      "300\t/471\t21.0s\n",
      "400\t/471\t24.0s\n",
      "471\t/471\t27.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t4\n",
      "Insignificant:\t47\n",
      "Filtered:\t0\n",
      "\n",
      "Total 2 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 27.0s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (4<500) significant events.\n",
      "Loading genome sequences ...\n",
      "Done, 63ms\n",
      "GC content=0.41\n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.51. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 51+/51- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=1\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 1\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "GAATTAATTC/GAATTAATTC\t1\t2\t0\t-1.5    \t0\t2.0\t0.0\t0.0\t1\n",
      "\n",
      "Align and cluster 1 k-mers with seed GAATTAATTC,   \t#0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= GAATTAATTC \t1.12, 2, kAUC=6.9\tPWM= GAAGTAGTGC\t9.67, 8, pAUC=13.7\n",
      "\n",
      "\n",
      "183ms: Finalizing 1 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out5_1 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out5_1 motif #0\n",
      "\n",
      "KSM top k-mer: GAATTAATTC/GAATTAATTC, total 1 k-mers\n",
      "KSM threshold: 1.12, \t\thit=2+/0-, kAUC=6.9\n",
      "\n",
      "                        |\n",
      "GAAGTAGTGC\n",
      "C T  GA TG\n",
      "          \n",
      "          \n",
      "PWM threshold: 9.67/13.76, \thit=8+/0-, pAUC=13.7\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "============================ Round 2 ============================\n",
      "Running EM with motif positional prior ...\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/471\t1.0s\n",
      "200\t/471\t10.0s\n",
      "300\t/471\t11.0s\n",
      "471\t/471\t11.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t5\n",
      "Insignificant:\t45\n",
      "Filtered:\t1\n",
      "\n",
      "Total 3 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 11.0s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (5<500) significant events.\n",
      "Percentage of significant events with a motif match, divided in 5 bins (~1 events per bin):\n",
      "0 100 0 0 0 \n",
      "\n",
      "Percentage of filtered events with a motif match, divided in 1 bins (~1 events per bin):\n",
      "0 \n",
      "\n",
      "Percentage of insignificant events with a motif match, divided in 9 bins (~5 events per bin):\n",
      "0 0 0 0 0 0 0 0 0 \n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.50. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 50+/50- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=1\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 1\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "GAATTAATTC/GAATTAATTC\t1\t2\t0\t-1.6    \t0\t2.0\t0.0\t0.0\t1\n",
      "\n",
      "Align and cluster 1 k-mers with seed GAATTAATTC,   \t#0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= GAATTAATTC \t1.12, 2, kAUC=6.9\tPWM= GAAGTAATGG\t7.85, 10, pAUC=14.2\n",
      "\n",
      "\n",
      "30ms: Finalizing 1 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out5_2 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out5_2 motif #0\n",
      "\n",
      "KSM top k-mer: GAATTAATTC/GAATTAATTC, total 1 k-mers\n",
      "KSM threshold: 1.12, \t\thit=2+/0-, kAUC=6.9\n",
      "\n",
      "                        |\n",
      "GAAGTAATGG\n",
      "     C  TC\n",
      "          \n",
      "          \n",
      "PWM threshold: 7.85/13.17, \thit=10+/3-, pAUC=14.2\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "Finished! GEM analysis results are printed to:\n",
      "/home/boycem4/chip_seq_project/DAY 4/out5/out5.GEM_events.txt\n",
      "/home/boycem4/chip_seq_project/DAY 4/out5/out5.results.htm\n",
      "/home/boycem4/chip_seq_project/DAY 4/out5/out5_outputs (folder with all other files)\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "End time: 2022/01/07 14:22:38\n",
      "\n",
      "Total running time: 56.8s\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#gem using outBED to load into IGV for chipA\n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar \\\n",
    "    --t 1 \\\n",
    "    --expt chipA.bam \\\n",
    "    --ctrl ctrlA.bam \\\n",
    "    --out out5 \\\n",
    "    --outBED \\\n",
    "    --f BAM \\\n",
    "    --d /usr/local/gem/Read_Distribution_default.txt \\\n",
    "    --genome . \\\n",
    "    --k 10 \\\n",
    "    --g chrom_sizes.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/11 19:12:31\n",
      "\n",
      "Loading data...\n",
      "    Loading reads from: chip.bam ... WARNING: BAM index file /home/boycem4/chip_seq_project/all/chip.bam.bai is older than BAM /home/boycem4/chip_seq_project/all/chip.bam\n",
      "Loaded\n",
      "    Loading reads from: ctrl.bam ... WARNING: BAM index file /home/boycem4/chip_seq_project/all/ctrl.bam.bai is older than BAM /home/boycem4/chip_seq_project/all/ctrl.bam\n",
      "Loaded\n",
      "\n",
      "Options:\n",
      "--t 1 --expt chip.bam --ctrl ctrl.bam --out out6 --outBED --f BAM --d /usr/local/gem/Read_Distribution_default.txt --genome . --k 10 --g chrom_sizes.txt \n",
      "\n",
      "Mappable Genome Length is 9,870,643.\n",
      "\n",
      "Getting 5' positions of all reads...\n",
      "\n",
      "Original read count stats:\n",
      "_IP  \tBases: 1215136\tHitCounts: 1634954.0\n",
      "_CTRL\tBases: 1107291\tHitCounts: 1673321.0\n",
      "\n",
      "Filter duplicate reads.\n",
      "_IP  \tBases: 1215136\tHitCounts: 1587966.0\n",
      "_CTRL\tBases: 1107291\tHitCounts: 1620135.0\n",
      "\n",
      "At Poisson p-value 1.0e-03, in a 501bp window, expect 110 reads.\n",
      "\n",
      "\n",
      "\tIP: 1587966\tCtrl: 1620135\t IP/Ctrl: 0.98\n",
      "\n",
      "Sorting reads and selecting enriched regions ...\n",
      "For condition , IP/Control = 2.70\n",
      "\n",
      "The genome is segmented into 10904 regions for analysis.\n",
      "\n",
      "============================ Round 0 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t29.0s\n",
      "1000\t/10904\t33.0s\n",
      "10904\t/10904\t33.0s\n",
      "\n",
      "Scaling condition , IP/Control = 1.84\n",
      "\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t51\n",
      "Insignificant:\t15\n",
      "Filtered:\t1\n",
      "\n",
      "Total 39 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 33.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (51<500) significant events.\n",
      "\n",
      "============================ Round 1 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t1.0s\n",
      "1000\t/10904\t2.0s\n",
      "10000\t/10904\t20.0s\n",
      "10904\t/10904\t21.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t59\n",
      "Insignificant:\t20\n",
      "Filtered:\t1\n",
      "\n",
      "Total 42 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 21.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (59<500) significant events.\n",
      "Loading genome sequences ...\n",
      "Done, 342ms\n",
      "GC content=0.41\n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.52. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 79+/79- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=13\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 13\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAA/TTCCGTGGAA\t6\t8\t0\t-3.5    \t0\t11.0\t8.0\t87.6\t5\n",
      "GGTTTCCGTG/CACGGAAACC\t3\t4\t0\t-2.2    \t2\t4.9\t4.0\t19.6\t4\n",
      "ATTTAAAGGG/CCCTTTAAAT\t3\t3\t0\t-1.8    \t6\t3.0\t4.0\t12.0\t3\n",
      "\n",
      "Align and cluster 8 k-mers with seed TTCCACGGAA,   \t#0\n",
      "Align and cluster 6 k-mers with seed GGTTTCCGTG,   \t#1\n",
      "Align and cluster 6 k-mers with seed ATTTAAAGGG,   \t#2\n",
      "\n",
      "Merging redundant motifs ...\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out6_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "\n",
      "------------------------- out6_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "------------------------- out6_1, k = ALL (after merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 15, kAUC=20.6\tPWM= TTCCACGGAA\t9.31, 32, pAUC=33.2\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 5, kAUC=9.8\tPWM= GGTTTCCGTG\t10.96, 18, pAUC=18.5\n",
      "#2\tk=10\tKSM= ATTTAAAGGG \t1.23, 3, kAUC=8.6\tPWM= ATTTAAAGGG\t13.57, 4, pAUC=6.0\n",
      "\n",
      "23ms: Finalizing 3 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out6_1 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out6_1 motif #0\n",
      "\n",
      "KSM top k-mer: TTCCACGGAA/TTCCGTGGAA, total 6 k-mers\n",
      "KSM threshold: 1.00, \t\thit=15+/0-, kAUC=20.6\n",
      "\n",
      "              |\n",
      "TTCCACGGAA\n",
      "          \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.31/17.47, \thit=32+/6-, pAUC=33.2\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out6_1 motif #1\n",
      "\n",
      "KSM top k-mer: GGTTTCCGTG/CACGGAAACC, total 4 k-mers\n",
      "KSM threshold: 1.11, \t\thit=5+/0-, kAUC=9.8\n",
      "\n",
      "GGTTTCCGTG\n",
      "C C\n",
      "PWM threshold: 10.96/17.07, \thit=18+/3-, pAUC=18.5\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out6_1 motif #2\n",
      "\n",
      "KSM top k-mer: ATTTAAAGGG/CCCTTTAAAT, total 4 k-mers\n",
      "KSM threshold: 1.23, \t\thit=3+/0-, kAUC=8.6\n",
      "\n",
      "          |\n",
      "ATTTAAAGGG\n",
      "  C  T    \n",
      "          \n",
      "          \n",
      "PWM threshold: 13.57/14.91, \thit=4+/0-, pAUC=6.0\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "============================ Round 2 ============================\n",
      "Running EM with motif positional prior ...\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10904\t1.0s\n",
      "1000\t/10904\t2.0s\n",
      "10000\t/10904\t26.0s\n",
      "10904\t/10904\t27.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t63\n",
      "Insignificant:\t22\n",
      "Filtered:\t0\n",
      "\n",
      "Total 46 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 27.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (63<500) significant events.\n",
      "Percentage of significant events with a motif match, divided in 9 bins (~7 events per bin):\n",
      "28 42 28 42 28 14 14 14 28 \n",
      "\n",
      "Percentage of insignificant events with a motif match, divided in 7 bins (~3 events per bin):\n",
      "0 0 0 0 0 0 0 0 \n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.52. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 85+/85- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=10 to k_max=10 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=16\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 16\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAA/TTCCGTGGAA\t8\t10\t0\t-4.1    \t0\t14.5\t8.0\t115.9\t8\n",
      "GGTGCTGCTC/GAGCAGCACC\t3\t3\t0\t-1.8    \t11\t3.0\t6.0\t18.0\t1\n",
      "GTGGGTTGTG/CACAACCCAC\t3\t3\t0\t-1.8    \t12\t3.0\t5.0\t15.0\t1\n",
      "GGTTTCCGTG/CACGGAAACC\t3\t4\t0\t-2.2    \t5\t5.3\t2.0\t10.6\t2\n",
      "CACTTCCACG/CGTGGAAGTG\t3\t3\t0\t-1.8    \t7\t3.0\t2.0\t6.0\t2\n",
      "\n",
      "Align and cluster 10 k-mers with seed TTCCACGGAA,   \t#0\n",
      "Align and cluster 1 k-mers with seed GGTGCTGCTC,   \t#1\n",
      "Align and cluster 4 k-mers with seed GTGGGTTGTG,   \t#2\n",
      "Align and cluster 6 k-mers with seed GGTTTCCGTG,   \t#3\n",
      "Align and cluster 8 k-mers with seed CACTTCCACG,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out6_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "\n",
      "------------------------- out6_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "------------------------- out6_2, k = ALL (after merging) -------------------------\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 19, kAUC=24.0\tPWM= TTCCACGGAA\t9.50, 33, pAUC=31.9\n",
      "#1\tk=10\tKSM= CACTTCCACG \t1.11, 16, kAUC=21.8\tPWM= CACTTCCACG\t9.55, 30, pAUC=27.5\n",
      "#2\tk=10\tKSM= GGTTTCCGTG \t1.11, 10, kAUC=15.1\tPWM= CGTTTCCGTG\t9.34, 31, pAUC=33.3\n",
      "#3\tk=10\tKSM= GTGGGTTGTG \t1.23, 3, kAUC=8.4\tPWM= GTGGTTTCTG\t10.94, 15, pAUC=16.4\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=8.4\tPWM= GGTGCTGCTC\t11.01, 12, pAUC=14.9\n",
      "\n",
      "24ms: Finalizing 5 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out6_2 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out6_2 motif #0\n",
      "\n",
      "KSM top k-mer: TTCCACGGAA/TTCCGTGGAA, total 9 k-mers\n",
      "KSM threshold: 1.00, \t\thit=19+/0-, kAUC=24.0\n",
      "\n",
      "              |\n",
      "TTCCACGGAA\n",
      "        G \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.50/17.74, \thit=33+/5-, pAUC=31.9\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out6_2 motif #1\n",
      "\n",
      "KSM top k-mer: CACTTCCACG/CGTGGAAGTG, total 8 k-mers\n",
      "KSM threshold: 1.11, \t\thit=16+/0-, kAUC=21.8\n",
      "\n",
      "                 |\n",
      "CACTTCCACG\n",
      "G GC   C  \n",
      "   A      \n",
      "          \n",
      "PWM threshold: 9.55/14.30, \thit=30+/5-, pAUC=27.5\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out6_2 motif #2\n",
      "\n",
      "KSM top k-mer: GGTTTCCGTG/CACGGAAACC, total 5 k-mers\n",
      "KSM threshold: 1.11, \t\thit=10+/0-, kAUC=15.1\n",
      "\n",
      "|\n",
      "CGTTTCCGTG\n",
      "GC        \n",
      "          \n",
      "          \n",
      "PWM threshold: 9.34/16.51, \thit=31+/2-, pAUC=33.3\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out6_2 motif #3\n",
      "\n",
      "KSM top k-mer: GTGGGTTGTG/CACAACCCAC, total 1 k-mers\n",
      "KSM threshold: 1.23, \t\thit=3+/0-, kAUC=8.4\n",
      "\n",
      "         |\n",
      "GTGGTTTCTG\n",
      "    G  GC \n",
      "          \n",
      "          \n",
      "PWM threshold: 10.94/15.74, \thit=15+/1-, pAUC=16.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out6_2 motif #4\n",
      "\n",
      "KSM top k-mer: GGTGCTGCTC/GAGCAGCACC, total 1 k-mers\n",
      "KSM threshold: 1.34, \t\thit=3+/0-, kAUC=8.4\n",
      "\n",
      "GGTGCTGCTC\n",
      "PWM threshold: 11.01/17.77, \thit=12+/1-, pAUC=14.9\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "Finished! GEM analysis results are printed to:\n",
      "/home/boycem4/chip_seq_project/all/out6/out6.GEM_events.txt\n",
      "/home/boycem4/chip_seq_project/all/out6/out6.results.htm\n",
      "/home/boycem4/chip_seq_project/all/out6/out6_outputs (folder with all other files)\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "End time: 2022/01/11 19:14:05\n",
      "\n",
      "Total running time: 1.6m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#outbed for whole genome\n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar \\\n",
    "    --t 1 \\\n",
    "    --expt chip.bam \\\n",
    "    --ctrl ctrl.bam \\\n",
    "    --out out6 \\\n",
    "    --outBED \\\n",
    "    --f BAM \\\n",
    "    --d /usr/local/gem/Read_Distribution_default.txt \\\n",
    "    --genome . \\\n",
    "    --k 10 \\\n",
    "    --g chrom_sizes.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/11 19:14:17\n",
      "\n",
      "Loading data...\n",
      "    Loading reads from: chip_q30.bam ... Loaded\n",
      "    Loading reads from: ctrl_q30.bam ... Loaded\n",
      "\n",
      "Options:\n",
      "--t 1 --expt chip_q30.bam --f BAM --genome . --g chrom_sizes.txt --d Read_Distribution_default.txt --k_min 8 --k_max 13 --ctrl ctrl_q30.bam --out out7 --outBED \n",
      "\n",
      "Mappable Genome Length is 9,870,643.\n",
      "\n",
      "Getting 5' positions of all reads...\n",
      "\n",
      "Original read count stats:\n",
      "_IP  \tBases: 1132522\tHitCounts: 1493105.0\n",
      "_CTRL\tBases: 1013379\tHitCounts: 1483523.0\n",
      "\n",
      "Filter duplicate reads.\n",
      "_IP  \tBases: 1132522\tHitCounts: 1448453.0\n",
      "_CTRL\tBases: 1013379\tHitCounts: 1433506.0\n",
      "\n",
      "At Poisson p-value 1.0e-03, in a 501bp window, expect 101 reads.\n",
      "\n",
      "\n",
      "\tIP: 1448453\tCtrl: 1433506\t IP/Ctrl: 1.01\n",
      "\n",
      "Sorting reads and selecting enriched regions ...\n",
      "For condition , IP/Control = 1.78\n",
      "\n",
      "The genome is segmented into 10785 regions for analysis.\n",
      "\n",
      "============================ Round 0 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10785\t26.0s\n",
      "1000\t/10785\t29.0s\n",
      "10000\t/10785\t30.0s\n",
      "10785\t/10785\t30.0s\n",
      "\n",
      "Scaling condition , IP/Control = 1.78\n",
      "\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t57\n",
      "Insignificant:\t25\n",
      "Filtered:\t1\n",
      "\n",
      "Total 39 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 30.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (57<500) significant events.\n",
      "\n",
      "============================ Round 1 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10785\t1.0s\n",
      "1000\t/10785\t2.0s\n",
      "10000\t/10785\t20.0s\n",
      "10785\t/10785\t22.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t57\n",
      "Insignificant:\t25\n",
      "Filtered:\t1\n",
      "\n",
      "Total 39 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 22.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (57<500) significant events.\n",
      "Loading genome sequences ...\n",
      "Done, 316ms\n",
      "GC content=0.41\n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.52. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 82+/82- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=8 to k_max=13 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=8 ...\n",
      "\n",
      "k=8, exact kmers=41\n",
      "k=8+1, gapped kmer=1, base kmer=2\n",
      "k=8+2, gapped kmer=0, base kmer=0\n",
      "k=8+3, gapped kmer=0, base kmer=0\n",
      "k=8+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 8 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 41\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.     \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCGTGG/CCACGGAA\t13\t15\t0\t-5.9    \t0\t20.5\t7.0\t143.5\t11\n",
      "CCCTCCCC/GGGGAGGG\t5\t6\t1\t-2.2    \t4\t9.8\t4.0\t39.2\t8\n",
      "CGGAACTG/CAGTTCCG\t3\t4\t0\t-2.2    \t13\t5.7\t4.0\t22.6\t3\n",
      "CGTGGAAT/ATTCCACG\t4\t5\t0\t-2.5    \t6\t8.7\t2.0\t17.3\t4\n",
      "GAAAGCCA/TGGCTTTC\t3\t4\t0\t-2.2    \t14\t4.0\t4.0\t16.0\t1\n",
      "\n",
      "Align and cluster 24 k-mers with seed TTCCGTGG,   \t#0\n",
      "Align and cluster 16 k-mers with seed CCCTCCCC,   \t#1\n",
      "Align and cluster 14 k-mers with seed CGGAACTG,   \t#2\n",
      "Align and cluster 22 k-mers with seed CGTGGAAT,   \t#3\n",
      "Align and cluster 4 k-mers with seed GAAAGCCA,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "17ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 8 ----------------------------\n",
      "\n",
      "#0\tk=8\tKSM= TTCCGTGG \t1.00, 30, kAUC=36.3\tPWM= TTCCGTGG\t10.78, 32, pAUC=36.6\n",
      "#1\tk=8\tKSM= CGTGGAAT \t1.00, 26, kAUC=31.6\tPWM= CGTGGAAG\t7.03, 33, pAUC=30.4\n",
      "#2\tk=8\tKSM= CGGAACTG \t1.11, 15, kAUC=21.2\tPWM= CGGAACTG\t9.13, 16, pAUC=17.0\n",
      "#3\tk=8\tKSM= CCCTCCCC \t1.23, 8, kAUC=16.5\tPWM= CCCCCCCC\t9.53, 13, pAUC=9.1\n",
      "#4\tk=8\tKSM= GAAAGCCA \t1.23, 7, kAUC=12.0\tPWM= GAAAGCCA\t8.78, 17, pAUC=13.8\n",
      "\n",
      "After merging:\n",
      "#0\tk=8\tKSM= TTCCGTGG \t1.00, 30, kAUC=36.3\tPWM= TTCCGTGG\t8.97, 43, pAUC=45.4\n",
      "#2\tk=8\tKSM= CGGAACTG \t1.11, 15, kAUC=21.2\tPWM= CGGAACTG\t9.13, 16, pAUC=17.0\n",
      "#3\tk=8\tKSM= CCCTCCCC \t1.23, 8, kAUC=16.5\tPWM= CCCCCCCC\t9.53, 13, pAUC=9.1\n",
      "#4\tk=8\tKSM= GAAAGCCA \t1.23, 7, kAUC=12.0\tPWM= GAAAGCCA\t8.78, 17, pAUC=13.8\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=9 ...\n",
      "\n",
      "k=9, exact kmers=17\n",
      "k=9+1, gapped kmer=1, base kmer=2\n",
      "k=9+2, gapped kmer=0, base kmer=0\n",
      "k=9+3, gapped kmer=0, base kmer=0\n",
      "k=9+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 9 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 17\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.       \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGA/TCCGTGGAA\t9\t11\t0\t-4.5    \t0\t14.5\t7.0\t101.2\t8\n",
      "CCCCTCCCC/GGGGAGGGG\t4\t5\t0\t-2.5    \t3\t5.0\t5.0\t25.0\t2\n",
      "GTTTCCGTG/CACGGAAAC\t3\t4\t0\t-2.2    \t7\t8.2\t2.0\t16.5\t3\n",
      "\n",
      "Align and cluster 12 k-mers with seed TTCCACGGA,   \t#0\n",
      "Align and cluster 2 k-mers with seed CCCCTCCCC,   \t#1\n",
      "Align and cluster 10 k-mers with seed GTTTCCGTG,   \t#2\n",
      "\n",
      "Merging redundant motifs ...\n",
      "10ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 9 ----------------------------\n",
      "\n",
      "#0\tk=9\tKSM= TTCCACGGA \t1.00, 22, kAUC=27.0\tPWM= TTCCACGGA\t10.07, 36, pAUC=39.5\n",
      "#1\tk=9\tKSM= GTTTCCGTG \t1.00, 18, kAUC=23.5\tPWM= GTTTCCGTG\t8.02, 34, pAUC=31.1\n",
      "#2\tk=9\tKSM= CCCCTCCCC \t1.11, 5, kAUC=9.6\tPWM= CCCCTCCCC\t10.33, 12, pAUC=11.7\n",
      "\n",
      "After merging:\n",
      "#0\tk=9\tKSM= TTCCACGGA \t1.00, 22, kAUC=27.0\tPWM= TTCCACGGA\t10.07, 36, pAUC=39.5\n",
      "#2\tk=9\tKSM= CCCCTCCCC \t1.11, 5, kAUC=9.6\tPWM= CCCCTCCCC\t10.33, 12, pAUC=11.7\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=8\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 8\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAA/TTCCGTGGAA\t6\t8\t0\t-3.5    \t0\t11.0\t7.0\t76.7\t6\n",
      "CCCCTCCCCG/CGGGGAGGGG\t3\t4\t0\t-2.2    \t1\t4.0\t6.0\t24.0\t1\n",
      "CGGATCCGTG/CACGGATCCG\t2\t2\t0\t-1.5    \t6\t2.0\t4.0\t8.0\t1\n",
      "\n",
      "Align and cluster 6 k-mers with seed TTCCACGGAA,   \t#0\n",
      "Align and cluster 2 k-mers with seed CCCCTCCCCG,   \t#1\n",
      "Align and cluster 3 k-mers with seed CGGATCCGTG,   \t#2\n",
      "\n",
      "Merging redundant motifs ...\n",
      "15ms: Remove motif #2.\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.11, 15, kAUC=20.1\tPWM= TTCCACGGAA\t11.54, 28, pAUC=33.0\n",
      "#1\tk=10\tKSM= CCCCTCCCCG \t1.34, 4, kAUC=8.5\tPWM= CCCCTCCCCG\t10.30, 16, pAUC=16.4\n",
      "#2\tk=10\tKSM= CGGATCCGTG \t1.23, 2, kAUC=7.3\tPWM= CGCTTCCGTG\t8.44, 31, pAUC=24.0\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.11, 15, kAUC=20.1\tPWM= TTCCACGGAA\t11.54, 28, pAUC=33.0\n",
      "#1\tk=10\tKSM= CCCCTCCCCG \t1.34, 4, kAUC=8.5\tPWM= CCCCTCCCCG\t10.30, 16, pAUC=16.4\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=11 ...\n",
      "\n",
      "k=11, exact kmers=2\n",
      "k=11+1, gapped kmer=0, base kmer=0\n",
      "k=11+2, gapped kmer=0, base kmer=0\n",
      "k=11+3, gapped kmer=0, base kmer=0\n",
      "k=11+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 11 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 2\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.           \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "CGGATCCGTGG/CCACGGATCCG\t2\t2\t0\t-1.5    \t0\t2.0\t2.0\t4.0\t2\n",
      "\n",
      "Align and cluster 2 k-mers with seed CGGATCCGTGG,   \t#0\n",
      "\n",
      "------------------------- k = 11 ----------------------------\n",
      "\n",
      "#0\tk=11\tKSM= CGGATCCGTGG \t1.23, 2, kAUC=7.3\tPWM= CGGTTCCGTGG\t8.79, 38, pAUC=45.1\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=12 ...\n",
      "\n",
      "k=12, exact kmers=1\n",
      "k=12+1, gapped kmer=0, base kmer=0\n",
      "k=12+2, gapped kmer=0, base kmer=0\n",
      "k=12+3, gapped kmer=0, base kmer=0\n",
      "k=12+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 12 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 1\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.             \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "CGGATCCGTGGA/TCCACGGATCCG\t2\t2\t0\t-1.5    \t0\t2.0\t0.0\t0.0\t1\n",
      "\n",
      "Align and cluster 1 k-mers with seed CGGATCCGTGGA,   \t#0\n",
      "\n",
      "------------------------- k = 12 ----------------------------\n",
      "\n",
      "#0\tk=12\tKSM= CGGATCCGTGGA \t1.23, 2, kAUC=7.3\tPWM= GGTTCCGTGG\t8.70, 34, pAUC=37.9\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=13 ...\n",
      "\n",
      "k=13, exact kmers=0\n",
      "k=13+1, gapped kmer=0, base kmer=0\n",
      "k=13+2, gapped kmer=0, base kmer=0\n",
      "k=13+3, gapped kmer=0, base kmer=0\n",
      "k=13+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "No enriched k-mer!\n",
      "\n",
      "------------------------- out7_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCGTGG \t1.00, 30, kAUC=36.3\tPWM= TTCCGTGG\t8.97, 43, pAUC=45.4\n",
      "#1\tk=9\tKSM= TTCCACGGA \t1.00, 22, kAUC=27.0\tPWM= TTCCACGGA\t10.07, 36, pAUC=39.5\n",
      "#2\tk=8\tKSM= CGGAACTG \t1.11, 15, kAUC=21.2\tPWM= CGGAACTG\t9.13, 16, pAUC=17.0\n",
      "#3\tk=10\tKSM= TTCCACGGAA \t1.11, 15, kAUC=20.1\tPWM= TTCCACGGAA\t11.54, 28, pAUC=33.0\n",
      "#4\tk=8\tKSM= CCCTCCCC \t1.23, 8, kAUC=16.5\tPWM= CCCCCCCC\t9.53, 13, pAUC=9.1\n",
      "#5\tk=8\tKSM= GAAAGCCA \t1.23, 7, kAUC=12.0\tPWM= GAAAGCCA\t8.78, 17, pAUC=13.8\n",
      "#6\tk=9\tKSM= CCCCTCCCC \t1.11, 5, kAUC=9.6\tPWM= CCCCTCCCC\t10.33, 12, pAUC=11.7\n",
      "#7\tk=10\tKSM= CCCCTCCCCG \t1.34, 4, kAUC=8.5\tPWM= CCCCTCCCCG\t10.30, 16, pAUC=16.4\n",
      "#8\tk=11\tKSM= CGGATCCGTGG \t1.23, 2, kAUC=7.3\tPWM= CGGTTCCGTGG\t8.79, 38, pAUC=45.1\n",
      "#9\tk=12\tKSM= CGGATCCGTGGA \t1.23, 2, kAUC=7.3\tPWM= GGTTCCGTGG\t8.70, 34, pAUC=37.9\n",
      "\n",
      "28ms: Remove motif #1.\n",
      "30ms: PWM TTCCGTGGAA has been removed.\n",
      "42ms: Remove motif #8.\n",
      "43ms: PWM GGTTCCGTGG has been removed.\n",
      "51ms: Remove motif #6.\n",
      "58ms: Remove motif #7.\n",
      "\n",
      "------------------------- out7_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCGTGG \t1.00, 30, kAUC=36.3\tPWM= TTCCGTGG\t8.97, 43, pAUC=45.4\n",
      "#1\tk=9\tKSM= TTCCACGGA \t1.00, 22, kAUC=27.0\tPWM= TTCCACGGA\t10.07, 36, pAUC=39.5\n",
      "#2\tk=8\tKSM= CGGAACTG \t1.11, 15, kAUC=21.2\tPWM= CGGAACTG\t9.13, 16, pAUC=17.0\n",
      "#3\tk=10\tKSM= TTCCACGGAA \t1.11, 15, kAUC=20.1\tPWM= TTCCACGGAA\t11.54, 28, pAUC=33.0\n",
      "#4\tk=8\tKSM= CCCTCCCC \t1.23, 8, kAUC=16.5\tPWM= CCCCCCCC\t9.53, 13, pAUC=9.1\n",
      "#5\tk=8\tKSM= GAAAGCCA \t1.23, 7, kAUC=12.0\tPWM= GAAAGCCA\t8.78, 17, pAUC=13.8\n",
      "#6\tk=9\tKSM= CCCCTCCCC \t1.11, 5, kAUC=9.6\tPWM= CCCCTCCCC\t10.33, 12, pAUC=11.7\n",
      "#7\tk=10\tKSM= CCCCTCCCCG \t1.34, 4, kAUC=8.5\tPWM= CCCCTCCCCG\t10.30, 16, pAUC=16.4\n",
      "#8\tk=11\tKSM= CGGATCCGTGG \t1.23, 2, kAUC=7.3\tPWM= CGGTTCCGTGG\t8.79, 38, pAUC=45.1\n",
      "#9\tk=12\tKSM= CGGATCCGTGGA \t1.23, 2, kAUC=7.3\tPWM= GGTTCCGTGG\t8.70, 34, pAUC=37.9\n",
      "\n",
      "------------------------- out7_1, k = ALL (after merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCGTGG \t1.00, 30, kAUC=36.3\tPWM= TTCCGTGG\t8.97, 43, pAUC=45.4\n",
      "#2\tk=8\tKSM= CGGAACTG \t1.11, 15, kAUC=21.2\tPWM= CGGAACTG\t9.13, 16, pAUC=17.0\n",
      "#4\tk=8\tKSM= CCCTCCCC \t1.29, 12, kAUC=18.9\tPWM= CCCCCCCC\t9.52, 14, pAUC=10.5\n",
      "#5\tk=8\tKSM= GAAAGCCA \t1.23, 7, kAUC=12.0\tPWM= GAAAGCCA\t8.78, 17, pAUC=13.8\n",
      "\n",
      "86ms: Finalizing 4 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out7_1 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out7_1 motif #0\n",
      "\n",
      "KSM top k-mer: TTCCGTGG/CCACGGAA, total 20 k-mers\n",
      "KSM threshold: 1.00, \t\thit=30+/0-, kAUC=36.3\n",
      "\n",
      "  |\n",
      "TTCCGTGG\n",
      "        \n",
      "        \n",
      "        \n",
      "PWM threshold: 8.97/14.96, \thit=43+/3-, pAUC=45.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_1 motif #1\n",
      "\n",
      "KSM top k-mer: CGGAACTG/CAGTTCCG, total 6 k-mers\n",
      "KSM threshold: 1.11, \t\thit=15+/0-, kAUC=21.2\n",
      "\n",
      "             |\n",
      "CGGAACTG\n",
      "        \n",
      "        \n",
      "        \n",
      "PWM threshold: 9.13/14.03, \thit=16+/2-, pAUC=17.0\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_1 motif #2\n",
      "\n",
      "KSM top k-mer: CCCTCCCC/GGGGAGGG, total 7 k-mers\n",
      "KSM threshold: 1.29, \t\thit=12+/0-, kAUC=18.9\n",
      "\n",
      "     |\n",
      "CCCCCCCC\n",
      "   TT   \n",
      "        \n",
      "        \n",
      "PWM threshold: 9.52/14.82, \thit=14+/9-, pAUC=10.5\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_1 motif #3\n",
      "\n",
      "KSM top k-mer: GAAAGCCA/TGGCTTTC, total 2 k-mers\n",
      "KSM threshold: 1.23, \t\thit=7+/0-, kAUC=12.0\n",
      "\n",
      "               |\n",
      "GAAAGCCA\n",
      " G    A \n",
      "        \n",
      "        \n",
      "PWM threshold: 8.78/12.54, \thit=17+/6-, pAUC=13.8\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "============================ Round 2 ============================\n",
      "Running EM with motif positional prior ...\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10785\t1.0s\n",
      "1000\t/10785\t2.0s\n",
      "10000\t/10785\t32.0s\n",
      "10785\t/10785\t33.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t68\n",
      "Insignificant:\t30\n",
      "Filtered:\t1\n",
      "\n",
      "Total 42 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 33.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (68<500) significant events.\n",
      "Percentage of significant events with a motif match, divided in 10 bins (~7 events per bin):\n",
      "85 71 71 57 57 57 28 71 100 20 \n",
      "\n",
      "Percentage of filtered events with a motif match, divided in 1 bins (~1 events per bin):\n",
      "0 \n",
      "\n",
      "Percentage of insignificant events with a motif match, divided in 10 bins (~3 events per bin):\n",
      "100 66 33 33 33 0 33 0 0 33 \n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.51. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 98+/98- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=8 to k_max=13 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=8 ...\n",
      "\n",
      "k=8, exact kmers=105\n",
      "k=8+1, gapped kmer=8, base kmer=17\n",
      "k=8+2, gapped kmer=5, base kmer=10\n",
      "k=8+3, gapped kmer=3, base kmer=7\n",
      "k=8+4, gapped kmer=2, base kmer=4\n",
      "\n",
      "------------------------- k = 8 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 105\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.     \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGG/CCGTGGAA\t24\t26\t0\t-9.8    \t0\t36.8\t8.0\t294.2\t15\n",
      "TTTCCGTG/CACGGAAA\t9\t12\t0\t-4.8    \t3\t18.7\t2.0\t37.3\t8\n",
      "AAACCATT/AATGGTTT\t4\t5\t0\t-2.5    \t14\t6.7\t4.0\t26.8\t2\n",
      "GTGGCTTC/GAAGCCAC\t3\t4\t0\t-2.2    \t56\t8.2\t3.0\t24.7\t5\n",
      "CGGAACAA/TTGTTCCG\t3\t3\t0\t-1.8    \t79\t6.0\t3.0\t18.0\t5\n",
      "\n",
      "Align and cluster 51 k-mers with seed TTCCACGG,   \t#0\n",
      "Align and cluster 53 k-mers with seed TTTCCGTG,   \t#1\n",
      "Align and cluster 28 k-mers with seed AAACCATT,   \t#2\n",
      "Align and cluster 28 k-mers with seed GTGGCTTC,   \t#3\n",
      "Align and cluster 36 k-mers with seed CGGAACAA,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "45ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 8 ----------------------------\n",
      "\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.00, 52, kAUC=53.0\tPWM= TTCCACGG\t9.16, 52, pAUC=49.5\n",
      "#1\tk=8\tKSM= TTTCCGTG \t1.00, 52, kAUC=51.6\tPWM= TTTCCGTG\t7.40, 48, pAUC=45.3\n",
      "#2\tk=8\tKSM= CGGAACAA \t1.11, 12, kAUC=18.5\tPWM= CGGAACAA\t7.60, 28, pAUC=16.8\n",
      "#3\tk=8\tKSM= GTGGCTTC \t1.00, 11, kAUC=15.7\tPWM= GTGGCTTC\t8.88, 18, pAUC=12.3\n",
      "#4\tk=8\tKSM= AAACCATT \t1.22, 12, kAUC=14.7\tPWM= AAACCACT\t7.81, 23, pAUC=17.0\n",
      "\n",
      "After merging:\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.00, 56, kAUC=56.0\tPWM= TTCCACGG\t8.85, 57, pAUC=50.2\n",
      "#2\tk=8\tKSM= CGGAACAA \t1.11, 12, kAUC=18.5\tPWM= CGGAACAA\t7.60, 28, pAUC=16.8\n",
      "#3\tk=8\tKSM= GTGGCTTC \t1.00, 11, kAUC=15.7\tPWM= GTGGCTTC\t8.88, 18, pAUC=12.3\n",
      "#4\tk=8\tKSM= AAACCATT \t1.22, 12, kAUC=14.7\tPWM= AAACCACT\t7.81, 23, pAUC=17.0\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=9 ...\n",
      "\n",
      "k=9, exact kmers=73\n",
      "k=9+1, gapped kmer=5, base kmer=10\n",
      "k=9+2, gapped kmer=2, base kmer=4\n",
      "k=9+3, gapped kmer=1, base kmer=2\n",
      "k=9+4, gapped kmer=1, base kmer=2\n",
      "\n",
      "------------------------- k = 9 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 73\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.       \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGA/TCCGTGGAA\t14\t16\t0\t-6.1    \t0\t25.6\t8.0\t204.9\t18\n",
      "CACGGAATG/CATTCCGTG\t4\t5\t0\t-2.5    \t9\t12.2\t2.0\t24.5\t8\n",
      "CATCGGACA/TGTCCGATG\t4\t4\t0\t-2.2    \t24\t7.5\t3.0\t22.4\t3\n",
      "ACAACCCAC/GTGGGTTGT\t4\t4\t0\t-2.2    \t17\t4.9\t4.0\t19.6\t2\n",
      "AGATCGTGG/CCACGATCT\t3\t3\t0\t-1.8    \t50\t5.7\t3.0\t17.2\t3\n",
      "\n",
      "Align and cluster 35 k-mers with seed TTCCACGGA,   \t#0\n",
      "Align and cluster 24 k-mers with seed CACGGAATG,   \t#1\n",
      "Align and cluster 14 k-mers with seed CATCGGACA,   \t#2\n",
      "Align and cluster 7 k-mers with seed ACAACCCAC,   \t#3\n",
      "Align and cluster 16 k-mers with seed AGATCGTGG,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "12ms: Remove motif #1.\n",
      "22ms: Remove motif #2.\n",
      "\n",
      "------------------------- k = 9 ----------------------------\n",
      "\n",
      "#0\tk=9\tKSM= TTCCACGGA \t1.00, 43, kAUC=44.7\tPWM= TTCCACGGA\t9.25, 57, pAUC=51.4\n",
      "#1\tk=9\tKSM= CACGGAATG \t1.00, 33, kAUC=34.1\tPWM= CACGGAAAC\t8.48, 55, pAUC=47.3\n",
      "#2\tk=9\tKSM= CATCGGACA \t1.00, 32, kAUC=34.1\tPWM= CACGGAA\t8.60, 57, pAUC=48.0\n",
      "#3\tk=9\tKSM= AGATCGTGG \t1.00, 7, kAUC=11.8\tPWM= AGGCCGTGG\t8.50, 24, pAUC=17.5\n",
      "#4\tk=9\tKSM= ACAACCCAC \t1.34, 4, kAUC=8.9\tPWM= ACAACCCAC\t10.55, 14, pAUC=12.9\n",
      "\n",
      "After merging:\n",
      "#0\tk=9\tKSM= TTCCACGGA \t1.00, 43, kAUC=44.7\tPWM= TTCCACGGA\t9.25, 57, pAUC=51.4\n",
      "#3\tk=9\tKSM= AGATCGTGG \t1.00, 7, kAUC=11.8\tPWM= AGGCCGTGG\t8.50, 24, pAUC=17.5\n",
      "#4\tk=9\tKSM= ACAACCCAC \t1.34, 4, kAUC=8.9\tPWM= ACAACCCAC\t10.55, 14, pAUC=12.9\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=59\n",
      "k=10+1, gapped kmer=1, base kmer=2\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 59\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAA/TTCCGTGGAA\t9\t11\t0\t-4.4    \t0\t18.2\t10.0\t181.7\t13\n",
      "AAAAGATCGT/ACGATCTTTT\t4\t4\t0\t-2.2    \t5\t4.0\t5.0\t20.0\t3\n",
      "ACAAAAGGTT/AACCTTTTGT\t4\t4\t0\t-2.2    \t7\t4.0\t5.0\t20.0\t3\n",
      "ATATTTTGCT/AGCAAAATAT\t4\t4\t0\t-2.2    \t9\t4.0\t5.0\t20.0\t3\n",
      "CCACGGCATC/GATGCCGTGG\t4\t4\t0\t-2.2    \t16\t8.7\t2.0\t17.4\t5\n",
      "\n",
      "Align and cluster 29 k-mers with seed TTCCACGGAA,   \t#0\n",
      "Align and cluster 12 k-mers with seed AAAAGATCGT,   \t#1\n",
      "Align and cluster 10 k-mers with seed ACAAAAGGTT,   \t#2\n",
      "Align and cluster 13 k-mers with seed ATATTTTGCT,   \t#3\n",
      "Align and cluster 29 k-mers with seed CCACGGCATC,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "16ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 32, kAUC=34.1\tPWM= TTCCACGGAA\t11.02, 47, pAUC=46.8\n",
      "#1\tk=10\tKSM= CCACGGCATC \t1.00, 32, kAUC=34.1\tPWM= CCACGGAATC\t10.30, 48, pAUC=41.7\n",
      "#2\tk=10\tKSM= ACAAAAGGTT \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTT\t9.91, 15, pAUC=15.4\n",
      "#3\tk=10\tKSM= AAAAGATCGT \t1.00, 4, kAUC=8.9\tPWM= AAAAGATCGT\t8.14, 9, pAUC=13.3\n",
      "#4\tk=10\tKSM= ATATTTTGCT \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCT\t11.53, 5, pAUC=7.5\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 32, kAUC=34.1\tPWM= TTCCACGGAA\t11.02, 47, pAUC=46.8\n",
      "#2\tk=10\tKSM= ACAAAAGGTT \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTT\t9.91, 15, pAUC=15.4\n",
      "#3\tk=10\tKSM= AAAAGATCGT \t1.00, 4, kAUC=8.9\tPWM= AAAAGATCGT\t8.14, 9, pAUC=13.3\n",
      "#4\tk=10\tKSM= ATATTTTGCT \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCT\t11.53, 5, pAUC=7.5\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=11 ...\n",
      "\n",
      "k=11, exact kmers=50\n",
      "k=11+1, gapped kmer=0, base kmer=0\n",
      "k=11+2, gapped kmer=0, base kmer=0\n",
      "k=11+3, gapped kmer=0, base kmer=0\n",
      "k=11+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 11 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 50\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.           \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAAA/TTTCCGTGGAA\t4\t5\t0\t-2.5    \t1\t10.0\t10.0\t100.0\t9\n",
      "AAAAGATCGTG/CACGATCTTTT\t4\t4\t0\t-2.2    \t2\t4.0\t7.0\t28.0\t2\n",
      "ATATTTTGCTT/AAGCAAAATAT\t4\t4\t0\t-2.2    \t5\t4.0\t7.0\t28.0\t3\n",
      "ACAAAAGGTTC/GAACCTTTTGT\t4\t4\t0\t-2.2    \t3\t4.0\t6.0\t24.0\t2\n",
      "ATCGGACAAAA/TTTTGTCCGAT\t4\t4\t0\t-2.2    \t6\t4.0\t5.0\t20.0\t3\n",
      "TCCACGGAATG/CATTCCGTGGA\t4\t5\t0\t-2.5\n",
      "\n",
      "Align and cluster 18 k-mers with seed TTCCACGGAAA,   \t#0\n",
      "Align and cluster 5 k-mers with seed AAAAGATCGTG,   \t#1\n",
      "Align and cluster 6 k-mers with seed ATATTTTGCTT,   \t#2\n",
      "Align and cluster 6 k-mers with seed ACAAAAGGTTC,   \t#3\n",
      "Align and cluster 10 k-mers with seed ATCGGACAAAA,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "4ms: PWM ATCGGACAAAA has been removed.\n",
      "\n",
      "------------------------- k = 11 ----------------------------\n",
      "\n",
      "#0\tk=11\tKSM= TTCCACGGAAA \t1.00, 23, kAUC=25.4\tPWM= TTCCACGGAAC\t9.86, 49, pAUC=48.3\n",
      "#1\tk=11\tKSM= ACAAAAGGTTC \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTTC\t11.34, 10, pAUC=14.9\n",
      "#2\tk=11\tKSM= AAAAGATCGTG \t1.00, 4, kAUC=8.9\tPWM= AAAAGACCGTG\t12.45, 7, pAUC=10.3\n",
      "#3\tk=11\tKSM= ATATTTTGCTT \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTT\t13.06, 5, pAUC=8.4\n",
      "#4\tk=11\tKSM= ATCGGACAAAA \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAA\t17.84, 4, pAUC=5.9\n",
      "\n",
      "After merging:\n",
      "#0\tk=11\tKSM= TTCCACGGAAA \t1.00, 23, kAUC=25.4\tPWM= TTCCACGGAAC\t9.86, 49, pAUC=48.3\n",
      "#1\tk=11\tKSM= ACAAAAGGTTC \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTTC\t11.34, 10, pAUC=14.9\n",
      "#2\tk=11\tKSM= AAAAGATCGTG \t1.00, 4, kAUC=8.9\tPWM= AAAAGACCGTG\t12.45, 7, pAUC=10.3\n",
      "#3\tk=11\tKSM= ATATTTTGCTT \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTT\t13.06, 5, pAUC=8.4\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=12 ...\n",
      "\n",
      "k=12, exact kmers=39\n",
      "k=12+1, gapped kmer=0, base kmer=0\n",
      "k=12+2, gapped kmer=0, base kmer=0\n",
      "k=12+3, gapped kmer=0, base kmer=0\n",
      "k=12+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 12 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 39\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.             \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "CACGGCATCGGA/TCCGATGCCGTG\t4\t4\t0\t-2.2    \t4\t5.3\t10.0\t52.9\t6\n",
      "GCTTCCACGGCA/TGCCGTGGAAGC\t4\t4\t0\t-2.2    \t13\t5.3\t8.0\t42.3\t6\n",
      "ATATTTTGCTTC/GAAGCAAAATAT\t4\t4\t0\t-2.2    \t1\t4.0\t8.0\t32.0\t5\n",
      "ATCGGACAAAAG/CTTTTGTCCGAT\t4\t4\t0\t-2.2    \t2\t4.0\t7.0\t28.0\t5\n",
      "AAAAGATCGTGG/CCACGATCTTTT\t3\t3\t0\t-1.8    \t24\t3.0\t6.0\t18.0\t3\n",
      "ACGGCATCGGAC/GTCCGATGCCGT\t4\t4\t0\t-2.2\n",
      "\n",
      "Align and cluster 12 k-mers with seed CACGGCATCGGA,   \t#0\n",
      "Align and cluster 10 k-mers with seed GCTTCCACGGCA,   \t#1\n",
      "Align and cluster 10 k-mers with seed ATATTTTGCTTC,   \t#2\n",
      "Align and cluster 10 k-mers with seed ATCGGACAAAAG,   \t#3\n",
      "Align and cluster 7 k-mers with seed AAAAGATCGTGG,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "19ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 12 ----------------------------\n",
      "\n",
      "#0\tk=12\tKSM= GCTTCCACGGCA \t1.34, 7, kAUC=11.8\tPWM= GCTTCCACGGAA\t12.32, 41, pAUC=43.5\n",
      "#1\tk=12\tKSM= CACGGCATCGGA \t1.34, 4, kAUC=8.9\tPWM= CACGGCATCGGA\t11.47, 15, pAUC=23.1\n",
      "#2\tk=12\tKSM= ATATTTTGCTTC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTC\t14.66, 5, pAUC=6.7\n",
      "#3\tk=12\tKSM= ATCGGACAAAAG \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAAG\t19.80, 4, pAUC=6.4\n",
      "#4\tk=12\tKSM= AAAAGATCGTGG \t1.22, 3, kAUC=7.9\tPWM= AAAAGATCGTGG\t9.79, 6, pAUC=12.0\n",
      "\n",
      "After merging:\n",
      "#0\tk=12\tKSM= GCTTCCACGGCA \t1.34, 7, kAUC=11.8\tPWM= GCTTCCACGGAA\t12.32, 41, pAUC=43.5\n",
      "#2\tk=12\tKSM= ATATTTTGCTTC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTC\t14.66, 5, pAUC=6.7\n",
      "#3\tk=12\tKSM= ATCGGACAAAAG \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAAG\t19.80, 4, pAUC=6.4\n",
      "#4\tk=12\tKSM= AAAAGATCGTGG \t1.22, 3, kAUC=7.9\tPWM= AAAAGATCGTGG\t9.79, 6, pAUC=12.0\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=13 ...\n",
      "\n",
      "k=13, exact kmers=36\n",
      "k=13+1, gapped kmer=0, base kmer=0\n",
      "k=13+2, gapped kmer=0, base kmer=0\n",
      "k=13+3, gapped kmer=0, base kmer=0\n",
      "k=13+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 13 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 36\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.               \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "CCACGGCATCGGA/TCCGATGCCGTGG\t4\t4\t0\t-2.2    \t6\t5.3\t11.0\t58.2\t6\n",
      "TTGCTTCCACGGC/GCCGTGGAAGCAA\t4\t4\t0\t-2.2    \t20\t5.3\t8.0\t42.3\t6\n",
      "ATATTTTGCTTCC/GGAAGCAAAATAT\t4\t4\t0\t-2.2    \t1\t4.0\t9.0\t36.0\t5\n",
      "ATCGGACAAAAGG/CCTTTTGTCCGAT\t4\t4\t0\t-2.2    \t2\t4.0\t7.0\t28.0\t5\n",
      "AAAAGATCGTGGG/CCCACGATCTTTT\t3\t3\t0\t-1.8    \t23\t3.0\t7.0\t21.0\t3\n",
      "ACGGCATCGGACA/TGTCCGATGCCGT\t4\t4\t0\t-2.2\n",
      "\n",
      "Align and cluster 9 k-mers with seed CCACGGCATCGGA,   \t#0\n",
      "Align and cluster 10 k-mers with seed TTGCTTCCACGGC,   \t#1\n",
      "Align and cluster 7 k-mers with seed ATATTTTGCTTCC,   \t#2\n",
      "Align and cluster 7 k-mers with seed ATCGGACAAAAGG,   \t#3\n",
      "Align and cluster 5 k-mers with seed AAAAGATCGTGGG,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "18ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 13 ----------------------------\n",
      "\n",
      "#0\tk=13\tKSM= TTGCTTCCACGGC \t1.34, 7, kAUC=11.8\tPWM= GCTTCCACGGAA\t10.55, 49, pAUC=47.6\n",
      "#1\tk=13\tKSM= CCACGGCATCGGA \t1.34, 7, kAUC=11.8\tPWM= CCACGGAATCGGA\t10.76, 33, pAUC=30.0\n",
      "#2\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTCC\t15.80, 4, pAUC=9.3\n",
      "#3\tk=13\tKSM= ATCGGACAAAAGG \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAAGG\t15.79, 4, pAUC=7.4\n",
      "#4\tk=13\tKSM= AAAAGATCGTGGG \t1.22, 3, kAUC=7.9\tPWM= AAAAGATCGTGGG\t19.56, 4, pAUC=11.2\n",
      "\n",
      "After merging:\n",
      "#0\tk=13\tKSM= TTGCTTCCACGGC \t1.34, 7, kAUC=11.8\tPWM= GCTTCCACGGAA\t10.55, 49, pAUC=47.6\n",
      "#2\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTCC\t15.80, 4, pAUC=9.3\n",
      "#3\tk=13\tKSM= ATCGGACAAAAGG \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAAGG\t15.79, 4, pAUC=7.4\n",
      "#4\tk=13\tKSM= AAAAGATCGTGGG \t1.22, 3, kAUC=7.9\tPWM= AAAAGATCGTGGG\t19.56, 4, pAUC=11.2\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out7_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.00, 56, kAUC=56.0\tPWM= TTCCACGG\t8.85, 57, pAUC=50.2\n",
      "#1\tk=9\tKSM= TTCCACGGA \t1.00, 43, kAUC=44.7\tPWM= TTCCACGGA\t9.25, 57, pAUC=51.4\n",
      "#2\tk=10\tKSM= TTCCACGGAA \t1.00, 32, kAUC=34.1\tPWM= TTCCACGGAA\t11.02, 47, pAUC=46.8\n",
      "#3\tk=11\tKSM= TTCCACGGAAA \t1.00, 23, kAUC=25.4\tPWM= TTCCACGGAAC\t9.86, 49, pAUC=48.3\n",
      "#4\tk=8\tKSM= CGGAACAA \t1.11, 12, kAUC=18.5\tPWM= CGGAACAA\t7.60, 28, pAUC=16.8\n",
      "#5\tk=8\tKSM= GTGGCTTC \t1.00, 11, kAUC=15.7\tPWM= GTGGCTTC\t8.88, 18, pAUC=12.3\n",
      "#6\tk=8\tKSM= AAACCATT \t1.22, 12, kAUC=14.7\tPWM= AAACCACT\t7.81, 23, pAUC=17.0\n",
      "#7\tk=13\tKSM= TTGCTTCCACGGC \t1.34, 7, kAUC=11.8\tPWM= GCTTCCACGGAA\t10.55, 49, pAUC=47.6\n",
      "#8\tk=12\tKSM= GCTTCCACGGCA \t1.34, 7, kAUC=11.8\tPWM= GCTTCCACGGAA\t12.32, 41, pAUC=43.5\n",
      "#9\tk=9\tKSM= AGATCGTGG \t1.00, 7, kAUC=11.8\tPWM= AGGCCGTGG\t8.50, 24, pAUC=17.5\n",
      "#10\tk=10\tKSM= ACAAAAGGTT \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTT\t9.91, 15, pAUC=15.4\n",
      "#11\tk=11\tKSM= ACAAAAGGTTC \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTTC\t11.34, 10, pAUC=14.9\n",
      "#12\tk=10\tKSM= AAAAGATCGT \t1.00, 4, kAUC=8.9\tPWM= AAAAGATCGT\t8.14, 9, pAUC=13.3\n",
      "#13\tk=9\tKSM= ACAACCCAC \t1.34, 4, kAUC=8.9\tPWM= ACAACCCAC\t10.55, 14, pAUC=12.9\n",
      "#14\tk=11\tKSM= AAAAGATCGTG \t1.00, 4, kAUC=8.9\tPWM= AAAAGACCGTG\t12.45, 7, pAUC=10.3\n",
      "#15\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTCC\t15.80, 4, pAUC=9.3\n",
      "#16\tk=11\tKSM= ATATTTTGCTT \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTT\t13.06, 5, pAUC=8.4\n",
      "#17\tk=10\tKSM= ATATTTTGCT \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCT\t11.53, 5, pAUC=7.5\n",
      "#18\tk=13\tKSM= ATCGGACAAAAGG \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAAGG\t15.79, 4, pAUC=7.4\n",
      "#19\tk=12\tKSM= ATATTTTGCTTC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTC\t14.66, 5, pAUC=6.7\n",
      "#20\tk=12\tKSM= ATCGGACAAAAG \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAAG\t19.80, 4, pAUC=6.4\n",
      "#21\tk=12\tKSM= AAAAGATCGTGG \t1.22, 3, kAUC=7.9\tPWM= AAAAGATCGTGG\t9.79, 6, pAUC=12.0\n",
      "#22\tk=13\tKSM= AAAAGATCGTGGG \t1.22, 3, kAUC=7.9\tPWM= AAAAGATCGTGGG\t19.56, 4, pAUC=11.2\n",
      "\n",
      "22ms: Remove motif #1.\n",
      "32ms: Remove motif #2.\n",
      "41ms: Remove motif #3.\n",
      "53ms: Remove motif #7.\n",
      "62ms: Remove motif #8.\n",
      "71ms: Remove motif #12.\n",
      "79ms: Remove motif #14.\n",
      "87ms: Remove motif #21.\n",
      "88ms: PWM AAAAGATCGTGGG has been removed.\n",
      "97ms: Remove motif #11.\n",
      "106ms: Remove motif #18.\n",
      "107ms: PWM ATCGGACAAAAG has been removed.\n",
      "117ms: Remove motif #16.\n",
      "126ms: Remove motif #17.\n",
      "136ms: Remove motif #19.\n",
      "\n",
      "------------------------- out7_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.00, 56, kAUC=56.0\tPWM= TTCCACGG\t8.85, 57, pAUC=50.2\n",
      "#1\tk=9\tKSM= TTCCACGGA \t1.00, 43, kAUC=44.7\tPWM= TTCCACGGA\t9.25, 57, pAUC=51.4\n",
      "#2\tk=10\tKSM= TTCCACGGAA \t1.00, 32, kAUC=34.1\tPWM= TTCCACGGAA\t11.02, 47, pAUC=46.8\n",
      "#3\tk=11\tKSM= TTCCACGGAAA \t1.00, 23, kAUC=25.4\tPWM= TTCCACGGAAC\t9.86, 49, pAUC=48.3\n",
      "#4\tk=8\tKSM= CGGAACAA \t1.11, 12, kAUC=18.5\tPWM= CGGAACAA\t7.60, 28, pAUC=16.8\n",
      "#5\tk=8\tKSM= GTGGCTTC \t1.00, 11, kAUC=15.7\tPWM= GTGGCTTC\t8.88, 18, pAUC=12.3\n",
      "#6\tk=8\tKSM= AAACCATT \t1.22, 12, kAUC=14.7\tPWM= AAACCACT\t7.81, 23, pAUC=17.0\n",
      "#7\tk=13\tKSM= TTGCTTCCACGGC \t1.34, 7, kAUC=11.8\tPWM= GCTTCCACGGAA\t10.55, 49, pAUC=47.6\n",
      "#8\tk=12\tKSM= GCTTCCACGGCA \t1.34, 7, kAUC=11.8\tPWM= GCTTCCACGGAA\t12.32, 41, pAUC=43.5\n",
      "#9\tk=9\tKSM= AGATCGTGG \t1.00, 7, kAUC=11.8\tPWM= AGGCCGTGG\t8.50, 24, pAUC=17.5\n",
      "#10\tk=10\tKSM= ACAAAAGGTT \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTT\t9.91, 15, pAUC=15.4\n",
      "#11\tk=11\tKSM= ACAAAAGGTTC \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTTC\t11.34, 10, pAUC=14.9\n",
      "#12\tk=10\tKSM= AAAAGATCGT \t1.00, 4, kAUC=8.9\tPWM= AAAAGATCGT\t8.14, 9, pAUC=13.3\n",
      "#13\tk=9\tKSM= ACAACCCAC \t1.34, 4, kAUC=8.9\tPWM= ACAACCCAC\t10.55, 14, pAUC=12.9\n",
      "#14\tk=11\tKSM= AAAAGATCGTG \t1.00, 4, kAUC=8.9\tPWM= AAAAGACCGTG\t12.45, 7, pAUC=10.3\n",
      "#15\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTCC\t15.80, 4, pAUC=9.3\n",
      "#16\tk=11\tKSM= ATATTTTGCTT \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTT\t13.06, 5, pAUC=8.4\n",
      "#17\tk=10\tKSM= ATATTTTGCT \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCT\t11.53, 5, pAUC=7.5\n",
      "#18\tk=13\tKSM= ATCGGACAAAAGG \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAAGG\t15.79, 4, pAUC=7.4\n",
      "#19\tk=12\tKSM= ATATTTTGCTTC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTC\t14.66, 5, pAUC=6.7\n",
      "#20\tk=12\tKSM= ATCGGACAAAAG \t1.34, 4, kAUC=8.9\tPWM= ATCGGACAAAAG\t19.80, 4, pAUC=6.4\n",
      "#21\tk=12\tKSM= AAAAGATCGTGG \t1.22, 3, kAUC=7.9\tPWM= AAAAGATCGTGG\t9.79, 6, pAUC=12.0\n",
      "#22\tk=13\tKSM= AAAAGATCGTGGG \t1.22, 3, kAUC=7.9\tPWM= AAAAGATCGTGGG\t19.56, 4, pAUC=11.2\n",
      "\n",
      "------------------------- out7_2, k = ALL (after merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.00, 56, kAUC=56.0\tPWM= TTCCACGG\t8.85, 57, pAUC=50.2\n",
      "#4\tk=8\tKSM= CGGAACAA \t1.11, 12, kAUC=18.5\tPWM= CGGAACAA\t7.60, 28, pAUC=16.8\n",
      "#5\tk=8\tKSM= GTGGCTTC \t1.00, 11, kAUC=15.7\tPWM= GTGGCTTC\t8.88, 18, pAUC=12.3\n",
      "#6\tk=8\tKSM= AAACCATT \t1.22, 12, kAUC=14.7\tPWM= AAACCACT\t7.81, 23, pAUC=17.0\n",
      "#9\tk=9\tKSM= AGATCGTGG \t1.00, 10, kAUC=14.7\tPWM= AGACCGTGG\t9.72, 19, pAUC=16.7\n",
      "#10\tk=10\tKSM= ACAAAAGGTT \t1.34, 4, kAUC=8.9\tPWM= ACCCAAGGTT\t9.91, 15, pAUC=15.4\n",
      "#13\tk=9\tKSM= ACAACCCAC \t1.34, 4, kAUC=8.9\tPWM= ACAACCCAC\t10.55, 14, pAUC=12.9\n",
      "#15\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 4, kAUC=8.9\tPWM= ATATTTTGCTTCC\t15.80, 4, pAUC=9.3\n",
      "\n",
      "186ms: Finalizing 8 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out7_2 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out7_2 motif #0\n",
      "\n",
      "KSM top k-mer: TTCCACGG/CCGTGGAA, total 43 k-mers\n",
      "KSM threshold: 1.00, \t\thit=56+/0-, kAUC=56.0\n",
      "\n",
      "         |\n",
      "TTCCACGG\n",
      " C      \n",
      "        \n",
      "        \n",
      "PWM threshold: 8.85/14.37, \thit=57+/6-, pAUC=50.2\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_2 motif #1\n",
      "\n",
      "KSM top k-mer: CGGAACAA/TTGTTCCG, total 6 k-mers\n",
      "KSM threshold: 1.11, \t\thit=12+/0-, kAUC=18.5\n",
      "\n",
      "    |\n",
      "CGGAACAA\n",
      "       G\n",
      "        \n",
      "        \n",
      "PWM threshold: 7.60/12.70, \thit=28+/9-, pAUC=16.8\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_2 motif #2\n",
      "\n",
      "KSM top k-mer: GTGGCTTC/GAAGCCAC, total 6 k-mers\n",
      "KSM threshold: 1.00, \t\thit=11+/0-, kAUC=15.7\n",
      "\n",
      "GTGGCTTC\n",
      "AC\n",
      "PWM threshold: 8.88/13.00, \thit=18+/8-, pAUC=12.3\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_2 motif #3\n",
      "\n",
      "KSM top k-mer: AAACCATT/AATGGTTT, total 3 k-mers\n",
      "KSM threshold: 1.22, \t\thit=12+/0-, kAUC=14.7\n",
      "\n",
      "AAACCACT\n",
      "  G   T\n",
      "PWM threshold: 7.81/12.28, \thit=23+/10-, pAUC=17.0\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_2 motif #4\n",
      "\n",
      "KSM top k-mer: AGATCGTGG/CCACGATCT, total 8 k-mers\n",
      "KSM threshold: 1.00, \t\thit=10+/0-, kAUC=14.7\n",
      "\n",
      "               |\n",
      "AGACCGTGG\n",
      "G GT     \n",
      "  C      \n",
      "         \n",
      "PWM threshold: 9.72/13.96, \thit=19+/6-, pAUC=16.7\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_2 motif #5\n",
      "\n",
      "KSM top k-mer: ACAAAAGGTT/AACCTTTTGT, total 4 k-mers\n",
      "KSM threshold: 1.34, \t\thit=4+/0-, kAUC=8.9\n",
      "\n",
      "ACCCAAGGTT\n",
      "  AACC\n",
      "PWM threshold: 9.91/15.23, \thit=15+/0-, pAUC=15.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_2 motif #6\n",
      "\n",
      "KSM top k-mer: ACAACCCAC/GTGGGTTGT, total 1 k-mers\n",
      "KSM threshold: 1.34, \t\thit=4+/0-, kAUC=8.9\n",
      "\n",
      "            |\n",
      "ACAACCCAC\n",
      "  C      \n",
      "         \n",
      "         \n",
      "PWM threshold: 10.55/15.49, \thit=14+/2-, pAUC=12.9\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out7_2 motif #7\n",
      "\n",
      "KSM top k-mer: ATATTTTGCTTCC/GGAAGCAAAATAT, total 7 k-mers\n",
      "KSM threshold: 1.00, \t\thit=4+/0-, kAUC=8.9\n",
      "\n",
      "                  |\n",
      "ATATTTTGCTTCC\n",
      "             \n",
      "             \n",
      "             \n",
      "PWM threshold: 15.80/20.62, \thit=4+/0-, pAUC=9.3\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "Finished! GEM analysis results are printed to:\n",
      "/home/boycem4/chip_seq_project/all/out7/out7.GEM_events.txt\n",
      "/home/boycem4/chip_seq_project/all/out7/out7.results.htm\n",
      "/home/boycem4/chip_seq_project/all/out7/out7_outputs (folder with all other files)\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "End time: 2022/01/11 19:15:58\n",
      "\n",
      "Total running time: 1.7m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#outbed whole genome with quality control trimming at 40\n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar \\\n",
    "    --t 1 \\\n",
    "    --expt chip_q30.bam \\\n",
    "    --f BAM --genome . \\\n",
    "    --g chrom_sizes.txt \\\n",
    "    --d Read_Distribution_default.txt \\\n",
    "    --k_min 8 --k_max 13 \\\n",
    "    --ctrl ctrl_q30.bam \\\n",
    "    --out out7 \\\n",
    "    --outBED "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "GEM (version 3.4)!\n",
      "\n",
      "Please cite: \n",
      "Yuchun Guo, Shaun Mahony, David K. Gifford (2012) PLoS Computational Biology 8(8): e1002638. \n",
      "High Resolution Genome Wide Binding Event Finding and Motif Discovery Reveals Transcription Factor Spatial Binding Constraints. \n",
      "doi:10.1371/journal.pcbi.1002638\n",
      "\n",
      "Gifford Laboratory at MIT (http://cgs.csail.mit.edu/gem/).\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "Start time: 2022/01/11 19:16:44\n",
      "\n",
      "Loading data...\n",
      "    Loading reads from: chip_q30_rmdup.bam ... Loaded\n",
      "    Loading reads from: ctrl_q30_rmdup.bam ... Loaded\n",
      "\n",
      "Options:\n",
      "--t 1 --expt chip_q30_rmdup.bam --f BAM --genome . --g chrom_sizes.txt --d Read_Distribution_default.txt --k_min 8 --k_max 13 --ctrl ctrl_q30_rmdup.bam --out out8 --outBED \n",
      "\n",
      "Mappable Genome Length is 9,870,643.\n",
      "\n",
      "Getting 5' positions of all reads...\n",
      "\n",
      "Original read count stats:\n",
      "_IP  \tBases: 1132522\tHitCounts: 1470045.0\n",
      "_CTRL\tBases: 1013379\tHitCounts: 1430493.0\n",
      "\n",
      "Filter duplicate reads.\n",
      "_IP  \tBases: 1132522\tHitCounts: 1433670.0\n",
      "_CTRL\tBases: 1013379\tHitCounts: 1395664.0\n",
      "\n",
      "At Poisson p-value 1.0e-03, in a 501bp window, expect 101 reads.\n",
      "\n",
      "\n",
      "\tIP: 1433670\tCtrl: 1395664\t IP/Ctrl: 1.03\n",
      "\n",
      "Sorting reads and selecting enriched regions ...\n",
      "For condition , IP/Control = 1.78\n",
      "\n",
      "The genome is segmented into 10782 regions for analysis.\n",
      "\n",
      "============================ Round 0 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10782\t25.0s\n",
      "1000\t/10782\t29.0s\n",
      "10000\t/10782\t30.0s\n",
      "10782\t/10782\t30.0s\n",
      "\n",
      "Scaling condition , IP/Control = 1.78\n",
      "\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t61\n",
      "Insignificant:\t28\n",
      "Filtered:\t1\n",
      "\n",
      "Total 44 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 30.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (61<500) significant events.\n",
      "\n",
      "============================ Round 1 ============================\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10782\t1.0s\n",
      "1000\t/10782\t2.0s\n",
      "10000\t/10782\t20.0s\n",
      "10782\t/10782\t22.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t60\n",
      "Insignificant:\t28\n",
      "Filtered:\t1\n",
      "\n",
      "Total 44 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 22.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (60<500) significant events.\n",
      "Loading genome sequences ...\n",
      "Done, 320ms\n",
      "GC content=0.41\n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.52. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 88+/88- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=8 to k_max=13 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=8 ...\n",
      "\n",
      "k=8, exact kmers=39\n",
      "k=8+1, gapped kmer=1, base kmer=2\n",
      "k=8+2, gapped kmer=0, base kmer=0\n",
      "k=8+3, gapped kmer=0, base kmer=0\n",
      "k=8+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 8 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 39\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.     \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGG/CCGTGGAA\t13\t15\t0\t-5.8    \t1\t20.5\t7.0\t143.5\t9\n",
      "CCCTCCCC/GGGGAGGG\t4\t5\t1\t-1.8    \t8\t6.7\t4.0\t26.8\t3\n",
      "TTTCCGTG/CACGGAAA\t8\t10\t0\t-4.1    \t3\t12.6\t2.0\t25.3\t4\n",
      "CTCCATTC/GAATGGAG\t3\t3\t0\t-1.8    \t21\t4.9\t3.0\t14.7\t3\n",
      "TCAGAGGA/TCCTCTGA\t3\t3\t0\t-1.8    \t33\t6.9\t2.0\t13.9\t1\n",
      "TTCCGTGG/CCACGGAA\t14\t16\t0\t-6.2\n",
      "\n",
      "Align and cluster 19 k-mers with seed TTCCACGG,   \t#0\n",
      "Align and cluster 13 k-mers with seed CCCTCCCC,   \t#1\n",
      "Align and cluster 19 k-mers with seed TTTCCGTG,   \t#2\n",
      "Align and cluster 11 k-mers with seed CTCCATTC,   \t#3\n",
      "Align and cluster 12 k-mers with seed TCAGAGGA,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "11ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 8 ----------------------------\n",
      "\n",
      "#0\tk=8\tKSM= TTTCCGTG \t1.11, 30, kAUC=33.4\tPWM= TTCCGT\t5.80, 35, pAUC=18.0\n",
      "#1\tk=8\tKSM= TTCCACGG \t1.11, 29, kAUC=33.3\tPWM= TTCCACGG\t9.39, 40, pAUC=37.5\n",
      "#2\tk=8\tKSM= CCCTCCCC \t1.22, 10, kAUC=15.0\tPWM= CCCCCCCC\t10.18, 14, pAUC=11.4\n",
      "#3\tk=8\tKSM= TCAGAGGA \t1.22, 5, kAUC=10.4\tPWM= TCCGCGGG\t8.31, 23, pAUC=20.7\n",
      "#4\tk=8\tKSM= CTCCATTC \t1.00, 4, kAUC=9.3\tPWM= CTCCATTC\t8.95, 15, pAUC=16.0\n",
      "\n",
      "After merging:\n",
      "#0\tk=8\tKSM= TTTCCGTG \t1.11, 34, kAUC=37.8\tPWM= TTCCGT\t6.85, 49, pAUC=15.4\n",
      "#2\tk=8\tKSM= CCCTCCCC \t1.22, 10, kAUC=15.0\tPWM= CCCCCCCC\t10.18, 14, pAUC=11.4\n",
      "#3\tk=8\tKSM= TCAGAGGA \t1.22, 5, kAUC=10.4\tPWM= TCCGCGGG\t8.31, 23, pAUC=20.7\n",
      "#4\tk=8\tKSM= CTCCATTC \t1.00, 4, kAUC=9.3\tPWM= CTCCATTC\t8.95, 15, pAUC=16.0\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=9 ...\n",
      "\n",
      "k=9, exact kmers=16\n",
      "k=9+1, gapped kmer=1, base kmer=2\n",
      "k=9+2, gapped kmer=0, base kmer=0\n",
      "k=9+3, gapped kmer=0, base kmer=0\n",
      "k=9+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 9 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 16\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.       \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGA/TCCGTGGAA\t10\t12\t0\t-4.8    \t0\t15.9\t7.0\t111.1\t8\n",
      "CCCCTCCCC/GGGGAGGGG\t3\t4\t1\t-1.5    \t6\t4.0\t5.0\t20.0\t1\n",
      "GTTTCCGTG/CACGGAAAC\t4\t5\t0\t-2.5    \t4\t9.5\t2.0\t19.0\t4\n",
      "\n",
      "Align and cluster 12 k-mers with seed TTCCACGGA,   \t#0\n",
      "Align and cluster 1 k-mers with seed CCCCTCCCC,   \t#1\n",
      "Align and cluster 10 k-mers with seed GTTTCCGTG,   \t#2\n",
      "\n",
      "Merging redundant motifs ...\n",
      "17ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 9 ----------------------------\n",
      "\n",
      "#0\tk=9\tKSM= TTCCACGGA \t1.00, 24, kAUC=27.7\tPWM= TCCCACGGA\t9.48, 39, pAUC=38.6\n",
      "#1\tk=9\tKSM= GTTTCCGTG \t1.00, 20, kAUC=23.6\tPWM= GTTTCCGTG\t10.07, 28, pAUC=28.8\n",
      "#2\tk=9\tKSM= CCCCTCCCC \t1.21, 4, kAUC=7.1\tPWM= CCCCTCCCC\t11.80, 9, pAUC=8.6\n",
      "\n",
      "After merging:\n",
      "#0\tk=9\tKSM= TTCCACGGA \t1.00, 24, kAUC=27.7\tPWM= TCCCACGGA\t9.48, 39, pAUC=38.6\n",
      "#2\tk=9\tKSM= CCCCTCCCC \t1.21, 4, kAUC=7.1\tPWM= CCCCTCCCC\t11.80, 9, pAUC=8.6\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=9\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 9\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAA/TTCCGTGGAA\t7\t9\t0\t-3.8    \t0\t12.0\t7.0\t84.0\t6\n",
      "GGTTTCCGTG/CACGGAAACC\t3\t4\t0\t-2.2    \t3\t4.9\t4.0\t19.6\t3\n",
      "\n",
      "Align and cluster 7 k-mers with seed TTCCACGGAA,   \t#0\n",
      "Align and cluster 4 k-mers with seed GGTTTCCGTG,   \t#1\n",
      "\n",
      "Merging redundant motifs ...\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.11, 17, kAUC=21.2\tPWM= TTCCACGGAA\t10.39, 32, pAUC=32.5\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 6, kAUC=10.4\tPWM= CGTTTCCGTG\t8.58, 29, pAUC=28.8\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.11, 17, kAUC=21.2\tPWM= TTCCACGGAA\t10.39, 32, pAUC=32.5\n",
      "#1\tk=10\tKSM= GGTTTCCGTG \t1.11, 6, kAUC=10.4\tPWM= CGTTTCCGTG\t8.58, 29, pAUC=28.8\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=11 ...\n",
      "\n",
      "k=11, exact kmers=5\n",
      "k=11+1, gapped kmer=0, base kmer=0\n",
      "k=11+2, gapped kmer=0, base kmer=0\n",
      "k=11+3, gapped kmer=0, base kmer=0\n",
      "k=11+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 11 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 5\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.           \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "ATTCCACGGAA/TTCCGTGGAAT\t3\t4\t0\t-2.2    \t0\t4.9\t7.0\t34.3\t2\n",
      "GGTTTCCGTGG/CCACGGAAACC\t3\t4\t0\t-2.2    \t1\t4.9\t6.0\t29.4\t2\n",
      "\n",
      "Align and cluster 3 k-mers with seed ATTCCACGGAA,   \t#0\n",
      "Align and cluster 4 k-mers with seed GGTTTCCGTGG,   \t#1\n",
      "\n",
      "Merging redundant motifs ...\n",
      "21ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 11 ----------------------------\n",
      "\n",
      "#0\tk=11\tKSM= ATTCCACGGAA \t1.22, 8, kAUC=12.6\tPWM= CTTCCACGGAA\t11.02, 33, pAUC=34.6\n",
      "#1\tk=11\tKSM= GGTTTCCGTGG \t1.22, 8, kAUC=12.6\tPWM= GGTTTCCGTGG\t10.24, 34, pAUC=34.4\n",
      "\n",
      "After merging:\n",
      "#0\tk=11\tKSM= ATTCCACGGAA \t1.22, 8, kAUC=12.6\tPWM= CTTCCACGGAA\t11.02, 33, pAUC=34.6\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=12 ...\n",
      "\n",
      "k=12, exact kmers=1\n",
      "k=12+1, gapped kmer=0, base kmer=0\n",
      "k=12+2, gapped kmer=0, base kmer=0\n",
      "k=12+3, gapped kmer=0, base kmer=0\n",
      "k=12+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 12 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 1\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.             \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "CGGATCCGTGGA/TCCACGGATCCG\t2\t2\t0\t-1.5    \t0\t2.0\t0.0\t0.0\t1\n",
      "\n",
      "Align and cluster 1 k-mers with seed CGGATCCGTGGA,   \t#0\n",
      "\n",
      "------------------------- k = 12 ----------------------------\n",
      "\n",
      "#0\tk=12\tKSM= CGGATCCGTGGA \t1.22, 2, kAUC=7.2\tPWM= TTCCGTGGGA\t9.13, 35, pAUC=35.7\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=13 ...\n",
      "\n",
      "k=13, exact kmers=0\n",
      "k=13+1, gapped kmer=0, base kmer=0\n",
      "k=13+2, gapped kmer=0, base kmer=0\n",
      "k=13+3, gapped kmer=0, base kmer=0\n",
      "k=13+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "No enriched k-mer!\n",
      "\n",
      "------------------------- out8_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=8\tKSM= TTTCCGTG \t1.11, 34, kAUC=37.8\tPWM= TTCCGT\t6.85, 49, pAUC=15.4\n",
      "#1\tk=9\tKSM= TTCCACGGA \t1.00, 24, kAUC=27.7\tPWM= TCCCACGGA\t9.48, 39, pAUC=38.6\n",
      "#2\tk=10\tKSM= TTCCACGGAA \t1.11, 17, kAUC=21.2\tPWM= TTCCACGGAA\t10.39, 32, pAUC=32.5\n",
      "#3\tk=8\tKSM= CCCTCCCC \t1.22, 10, kAUC=15.0\tPWM= CCCCCCCC\t10.18, 14, pAUC=11.4\n",
      "#4\tk=11\tKSM= ATTCCACGGAA \t1.22, 8, kAUC=12.6\tPWM= CTTCCACGGAA\t11.02, 33, pAUC=34.6\n",
      "#5\tk=10\tKSM= GGTTTCCGTG \t1.11, 6, kAUC=10.4\tPWM= CGTTTCCGTG\t8.58, 29, pAUC=28.8\n",
      "#6\tk=8\tKSM= TCAGAGGA \t1.22, 5, kAUC=10.4\tPWM= TCCGCGGG\t8.31, 23, pAUC=20.7\n",
      "#7\tk=8\tKSM= CTCCATTC \t1.00, 4, kAUC=9.3\tPWM= CTCCATTC\t8.95, 15, pAUC=16.0\n",
      "#8\tk=12\tKSM= CGGATCCGTGGA \t1.22, 2, kAUC=7.2\tPWM= TTCCGTGGGA\t9.13, 35, pAUC=35.7\n",
      "#9\tk=9\tKSM= CCCCTCCCC \t1.21, 4, kAUC=7.1\tPWM= CCCCTCCCC\t11.80, 9, pAUC=8.6\n",
      "\n",
      "19ms: Remove motif #1.\n",
      "27ms: Remove motif #2.\n",
      "36ms: Remove motif #4.\n",
      "45ms: Remove motif #5.\n",
      "56ms: Remove motif #8.\n",
      "65ms: Remove motif #9.\n",
      "\n",
      "------------------------- out8_1, k = ALL (before merging) -------------------------\n",
      "#0\tk=8\tKSM= TTTCCGTG \t1.11, 34, kAUC=37.8\tPWM= TTCCGT\t6.85, 49, pAUC=15.4\n",
      "#1\tk=9\tKSM= TTCCACGGA \t1.00, 24, kAUC=27.7\tPWM= TCCCACGGA\t9.48, 39, pAUC=38.6\n",
      "#2\tk=10\tKSM= TTCCACGGAA \t1.11, 17, kAUC=21.2\tPWM= TTCCACGGAA\t10.39, 32, pAUC=32.5\n",
      "#3\tk=8\tKSM= CCCTCCCC \t1.22, 10, kAUC=15.0\tPWM= CCCCCCCC\t10.18, 14, pAUC=11.4\n",
      "#4\tk=11\tKSM= ATTCCACGGAA \t1.22, 8, kAUC=12.6\tPWM= CTTCCACGGAA\t11.02, 33, pAUC=34.6\n",
      "#5\tk=10\tKSM= GGTTTCCGTG \t1.11, 6, kAUC=10.4\tPWM= CGTTTCCGTG\t8.58, 29, pAUC=28.8\n",
      "#6\tk=8\tKSM= TCAGAGGA \t1.22, 5, kAUC=10.4\tPWM= TCCGCGGG\t8.31, 23, pAUC=20.7\n",
      "#7\tk=8\tKSM= CTCCATTC \t1.00, 4, kAUC=9.3\tPWM= CTCCATTC\t8.95, 15, pAUC=16.0\n",
      "#8\tk=12\tKSM= CGGATCCGTGGA \t1.22, 2, kAUC=7.2\tPWM= TTCCGTGGGA\t9.13, 35, pAUC=35.7\n",
      "#9\tk=9\tKSM= CCCCTCCCC \t1.21, 4, kAUC=7.1\tPWM= CCCCTCCCC\t11.80, 9, pAUC=8.6\n",
      "\n",
      "------------------------- out8_1, k = ALL (after merging) -------------------------\n",
      "#0\tk=8\tKSM= TTTCCGTG \t1.11, 34, kAUC=37.8\tPWM= TTCCGT\t6.85, 49, pAUC=15.4\n",
      "#3\tk=8\tKSM= CCCTCCCC \t1.22, 10, kAUC=15.0\tPWM= CCCCCCCC\t10.18, 14, pAUC=11.4\n",
      "#6\tk=8\tKSM= TCAGAGGA \t1.22, 5, kAUC=10.4\tPWM= TCCGCGGG\t8.31, 23, pAUC=20.7\n",
      "#7\tk=8\tKSM= CTCCATTC \t1.00, 4, kAUC=9.3\tPWM= CTCCATTC\t8.95, 15, pAUC=16.0\n",
      "\n",
      "80ms: Finalizing 4 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out8_1 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out8_1 motif #0\n",
      "\n",
      "KSM top k-mer: TTTCCGTG/CACGGAAA, total 19 k-mers\n",
      "KSM threshold: 1.11, \t\thit=34+/0-, kAUC=37.8\n",
      "\n",
      "TTCCGT\n",
      " C   G\n",
      "PWM threshold: 6.85/9.99, \thit=49+/25-, pAUC=15.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_1 motif #1\n",
      "\n",
      "KSM top k-mer: CCCTCCCC/GGGGAGGG, total 5 k-mers\n",
      "KSM threshold: 1.22, \t\thit=10+/1-, kAUC=15.0\n",
      "\n",
      "          |\n",
      "CCCCCCCC\n",
      "     G  \n",
      "        \n",
      "        \n",
      "PWM threshold: 10.18/15.47, \thit=14+/6-, pAUC=11.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_1 motif #2\n",
      "\n",
      "KSM top k-mer: TCAGAGGA/TCCTCTGA, total 2 k-mers\n",
      "KSM threshold: 1.22, \t\thit=5+/0-, kAUC=10.4\n",
      "\n",
      "TCCGCGGG\n",
      "  A A  A\n",
      "PWM threshold: 8.31/12.57, \thit=23+/7-, pAUC=20.7\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_1 motif #3\n",
      "\n",
      "KSM top k-mer: CTCCATTC/GAATGGAG, total 2 k-mers\n",
      "KSM threshold: 1.00, \t\thit=4+/0-, kAUC=9.3\n",
      "\n",
      "     |\n",
      "CTCCATTC\n",
      "       T\n",
      "        \n",
      "        \n",
      "PWM threshold: 8.95/13.47, \thit=15+/2-, pAUC=16.0\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "============================ Round 2 ============================\n",
      "Running EM with motif positional prior ...\n",
      "\n",
      "Running with 1 threads ...\n",
      "\n",
      "100\t/10782\t1.0s\n",
      "1000\t/10782\t2.0s\n",
      "10000\t/10782\t37.0s\n",
      "10782\t/10782\t38.0s\n",
      "---------------------------\n",
      "Events discovered \n",
      "Significant:\t73\n",
      "Insignificant:\t30\n",
      "Filtered:\t0\n",
      "\n",
      "Total 53 homotypic events (within 501bp of other events).\n",
      "\n",
      "Finish binding event prediction: 38.1s\n",
      "\n",
      "\n",
      "Warning: The read distribution has not been updated, too few (73<500) significant events.\n",
      "Percentage of significant events with a motif match, divided in 9 bins (~8 events per bin):\n",
      "62 62 62 50 87 50 50 50 87 0 \n",
      "\n",
      "Percentage of insignificant events with a motif match, divided in 10 bins (~3 events per bin):\n",
      "33 33 66 0 33 33 0 33 0 0 \n",
      "\n",
      "Use di-nucleotide shuffled sequences as negative sequences.\n",
      "\n",
      "Estimated GC content is 0.51. Set [--gc -1] to use the estimated GC content.\n",
      "Provided  GC content is 0.41.\n",
      "\n",
      "KMAC motif discovery with 103+/103- sequences.\n",
      "\n",
      "\n",
      "Motif discovery from k_min=8 to k_max=13 ...\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=8 ...\n",
      "\n",
      "k=8, exact kmers=101\n",
      "k=8+1, gapped kmer=7, base kmer=15\n",
      "k=8+2, gapped kmer=5, base kmer=10\n",
      "k=8+3, gapped kmer=4, base kmer=8\n",
      "k=8+4, gapped kmer=2, base kmer=4\n",
      "\n",
      "------------------------- k = 8 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 101\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.     \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGG/CCGTGGAA\t25\t27\t0\t-10.1    \t0\t37.1\t8.0\t296.9\t15\n",
      "TTTCCGTG/CACGGAAA\t8\t10\t0\t-4.1    \t3\t17.6\t2.0\t35.2\t8\n",
      "TCCTTTCC/GGAAAGGA\t4\t5\t0\t-2.5    \t20\t8.1\t4.0\t32.2\t6\n",
      "ACGGCATC/GATGCCGT\t5\t5\t0\t-2.5    \t12\t8.7\t2.0\t17.3\t4\n",
      "TCCACAAA/TTTGTGGA\t3\t3\t0\t-1.8    \t84\t8.5\t2.0\t17.0\t1\n",
      "\n",
      "Align and cluster 54 k-mers with seed TTCCACGG,   \t#0\n",
      "Align and cluster 55 k-mers with seed TTTCCGTG,   \t#1\n",
      "Align and cluster 35 k-mers with seed TCCTTTCC,   \t#2\n",
      "Align and cluster 32 k-mers with seed ACGGCATC,   \t#3\n",
      "Align and cluster 35 k-mers with seed TCCACAAA,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "26ms: Remove motif #1.\n",
      "40ms: Remove motif #2.\n",
      "\n",
      "------------------------- k = 8 ----------------------------\n",
      "\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.45, 41, kAUC=53.6\tPWM= TTCCACGG\t8.21, 61, pAUC=54.6\n",
      "#1\tk=8\tKSM= TCCACAAA \t1.22, 47, kAUC=51.0\tPWM= TTCCACGG\t8.92, 50, pAUC=41.3\n",
      "#2\tk=8\tKSM= TTTCCGTG \t1.52, 30, kAUC=46.8\tPWM= TTCCGT\t7.06, 48, pAUC=19.2\n",
      "#3\tk=8\tKSM= ACGGCATC \t1.11, 24, kAUC=25.8\tPWM= ACGGCATC\t8.27, 31, pAUC=19.1\n",
      "#4\tk=8\tKSM= TCCTTTCC \t1.11, 14, kAUC=18.0\tPWM= CCCTTTCC\t8.48, 26, pAUC=16.7\n",
      "\n",
      "After merging:\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.45, 41, kAUC=53.6\tPWM= TTCCACGG\t8.21, 61, pAUC=54.6\n",
      "#3\tk=8\tKSM= ACGGCATC \t1.11, 24, kAUC=25.8\tPWM= ACGGCATC\t8.27, 31, pAUC=19.1\n",
      "#4\tk=8\tKSM= TCCTTTCC \t1.11, 14, kAUC=18.0\tPWM= CCCTTTCC\t8.48, 26, pAUC=16.7\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=9 ...\n",
      "\n",
      "k=9, exact kmers=74\n",
      "k=9+1, gapped kmer=5, base kmer=10\n",
      "k=9+2, gapped kmer=0, base kmer=0\n",
      "k=9+3, gapped kmer=1, base kmer=2\n",
      "k=9+4, gapped kmer=1, base kmer=2\n",
      "\n",
      "------------------------- k = 9 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 74\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.       \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGA/TCCGTGGAA\t14\t16\t0\t-6.1    \t0\t24.7\t8.0\t197.3\t16\n",
      "GTTTCCGTG/CACGGAAAC\t3\t4\t0\t-2.2    \t15\t10.8\t2.0\t21.5\t8\n",
      "CATACCACC/GGTGGTATG\t3\t4\t0\t-2.2    \t10\t6.3\t3.0\t19.0\t1\n",
      "CCCCTCCCC/GGGGAGGGG\t3\t4\t0\t-2.2    \t11\t6.3\t3.0\t19.0\t3\n",
      "CAAATTGCA/TGCAATTTG\t3\t3\t0\t-1.8    \t33\t4.6\t3.0\t13.7\t4\n",
      "\n",
      "Align and cluster 32 k-mers with seed TTCCACGGA,   \t#0\n",
      "Align and cluster 26 k-mers with seed GTTTCCGTG,   \t#1\n",
      "Align and cluster 14 k-mers with seed CATACCACC,   \t#2\n",
      "Align and cluster 6 k-mers with seed CCCCTCCCC,   \t#3\n",
      "Align and cluster 12 k-mers with seed CAAATTGCA,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "14ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 9 ----------------------------\n",
      "\n",
      "#0\tk=9\tKSM= TTCCACGGA \t1.11, 33, kAUC=38.0\tPWM= TTCCACGGA\t8.40, 62, pAUC=55.6\n",
      "#1\tk=9\tKSM= GTTTCCGTG \t1.34, 24, kAUC=33.8\tPWM= GTTTCCGTG\t7.92, 65, pAUC=48.5\n",
      "#2\tk=9\tKSM= CAAATTGCA \t1.00, 4, kAUC=9.6\tPWM= CAAATTGCA\t7.92, 15, pAUC=14.6\n",
      "#3\tk=9\tKSM= CCCCTCCCC \t1.34, 6, kAUC=9.6\tPWM= CCCCCCCCC\t8.83, 20, pAUC=12.3\n",
      "#4\tk=9\tKSM= CATACCACC \t1.34, 4, kAUC=7.8\tPWM= CATACCACC\t10.22, 7, pAUC=7.1\n",
      "\n",
      "After merging:\n",
      "#0\tk=9\tKSM= TTCCACGGA \t1.11, 33, kAUC=38.0\tPWM= TTCCACGGA\t8.40, 62, pAUC=55.6\n",
      "#2\tk=9\tKSM= CAAATTGCA \t1.00, 4, kAUC=9.6\tPWM= CAAATTGCA\t7.92, 15, pAUC=14.6\n",
      "#3\tk=9\tKSM= CCCCTCCCC \t1.34, 6, kAUC=9.6\tPWM= CCCCCCCCC\t8.83, 20, pAUC=12.3\n",
      "#4\tk=9\tKSM= CATACCACC \t1.34, 4, kAUC=7.8\tPWM= CATACCACC\t10.22, 7, pAUC=7.1\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=10 ...\n",
      "\n",
      "k=10, exact kmers=59\n",
      "k=10+1, gapped kmer=0, base kmer=0\n",
      "k=10+2, gapped kmer=0, base kmer=0\n",
      "k=10+3, gapped kmer=0, base kmer=0\n",
      "k=10+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 59\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.         \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "TTCCACGGAA/TTCCGTGGAA\t9\t11\t0\t-4.4    \t0\t18.2\t10.0\t181.7\t14\n",
      "CGGATCCGTG/CACGGATCCG\t3\t3\t0\t-1.8    \t29\t4.9\t4.0\t19.6\t3\n",
      "ACAAAAGGTT/AACCTTTTGT\t4\t3\t0\t-1.8    \t10\t3.0\t5.0\t15.0\t3\n",
      "GGTGCTGCTC/GAGCAGCACC\t3\t3\t0\t-1.8    \t44\t3.0\t5.0\t15.0\t1\n",
      "CCACTTCCAC/GTGGAAGTGG\t3\t3\t0\t-1.8    \t27\t4.9\t3.0\t14.7\t2\n",
      "\n",
      "Align and cluster 27 k-mers with seed TTCCACGGAA,   \t#0\n",
      "Align and cluster 16 k-mers with seed CGGATCCGTG,   \t#1\n",
      "Align and cluster 11 k-mers with seed ACAAAAGGTT,   \t#2\n",
      "Align and cluster 4 k-mers with seed GGTGCTGCTC,   \t#3\n",
      "Align and cluster 15 k-mers with seed CCACTTCCAC,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "24ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 10 ----------------------------\n",
      "\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 32, kAUC=32.7\tPWM= TTCCACGGAA\t10.29, 52, pAUC=49.7\n",
      "#1\tk=10\tKSM= CGGATCCGTG \t1.00, 19, kAUC=21.6\tPWM= CGGTTCCGTG\t8.64, 48, pAUC=40.6\n",
      "#2\tk=10\tKSM= CCACTTCCAC \t1.22, 6, kAUC=9.6\tPWM= CCACTTCCAC\t10.28, 22, pAUC=17.0\n",
      "#3\tk=10\tKSM= ACAAAAGGTT \t1.34, 3, kAUC=8.7\tPWM= ACAAAAGGTT\t15.89, 3, pAUC=3.9\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=7.8\tPWM= GGTGCTGCTC\t10.75, 11, pAUC=12.3\n",
      "\n",
      "After merging:\n",
      "#0\tk=10\tKSM= TTCCACGGAA \t1.00, 35, kAUC=35.4\tPWM= TTCCACGGAA\t9.89, 59, pAUC=53.3\n",
      "#2\tk=10\tKSM= CCACTTCCAC \t1.22, 6, kAUC=9.6\tPWM= CCACTTCCAC\t10.28, 22, pAUC=17.0\n",
      "#3\tk=10\tKSM= ACAAAAGGTT \t1.34, 3, kAUC=8.7\tPWM= ACAAAAGGTT\t15.89, 3, pAUC=3.9\n",
      "#4\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=7.8\tPWM= GGTGCTGCTC\t10.75, 11, pAUC=12.3\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=11 ...\n",
      "\n",
      "k=11, exact kmers=45\n",
      "k=11+1, gapped kmer=0, base kmer=0\n",
      "k=11+2, gapped kmer=0, base kmer=0\n",
      "k=11+3, gapped kmer=0, base kmer=0\n",
      "k=11+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 11 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 45\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.           \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "ATTCCACGGAA/TTCCGTGGAAT\t4\t5\t0\t-2.5    \t0\t8.9\t10.0\t89.4\t6\n",
      "ACAAAAGGTTC/GAACCTTTTGT\t4\t3\t0\t-1.8    \t2\t3.0\t6.0\t18.0\t2\n",
      "ATATTTTGCTT/AAGCAAAATAT\t4\t3\t0\t-1.8    \t6\t3.0\t5.0\t15.0\t3\n",
      "CGTGGGATATT/AATATCCCACG\t3\t3\t0\t-1.8    \t19\t3.0\t5.0\t15.0\t2\n",
      "TCCACAAATTG/CAATTTGTGGA\t3\t3\t0\t-1.8    \t32\t4.2\t3.0\t12.7\t3\n",
      "\n",
      "Align and cluster 14 k-mers with seed ATTCCACGGAA,   \t#0\n",
      "Align and cluster 8 k-mers with seed ACAAAAGGTTC,   \t#1\n",
      "Align and cluster 7 k-mers with seed ATATTTTGCTT,   \t#2\n",
      "Align and cluster 6 k-mers with seed CGTGGGATATT,   \t#3\n",
      "Align and cluster 10 k-mers with seed TCCACAAATTG,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "3ms: PWM TCCACGGAAAG has been removed.\n",
      "12ms: Remove motif #2.\n",
      "\n",
      "------------------------- k = 11 ----------------------------\n",
      "\n",
      "#0\tk=11\tKSM= ATTCCACGGAA \t1.00, 18, kAUC=20.7\tPWM= TTCCACGGA\t9.07, 57, pAUC=53.5\n",
      "#1\tk=11\tKSM= TCCACAAATTG \t1.00, 10, kAUC=14.2\tPWM= TCCACGGAAAG\t9.17, 53, pAUC=49.4\n",
      "#2\tk=11\tKSM= CGTGGGATATT \t1.00, 7, kAUC=11.5\tPWM= TCCGTGGAA\t8.70, 58, pAUC=52.9\n",
      "#3\tk=11\tKSM= ATATTTTGCTT \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTT\t13.11, 5, pAUC=8.8\n",
      "#4\tk=11\tKSM= ACAAAAGGTTC \t1.34, 3, kAUC=8.7\tPWM= ACAAAAGGTTC\t13.04, 6, pAUC=5.7\n",
      "\n",
      "After merging:\n",
      "#0\tk=11\tKSM= ATTCCACGGAA \t1.00, 18, kAUC=20.7\tPWM= TTCCACGGA\t9.07, 57, pAUC=53.5\n",
      "#3\tk=11\tKSM= ATATTTTGCTT \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTT\t13.11, 5, pAUC=8.8\n",
      "#4\tk=11\tKSM= ACAAAAGGTTC \t1.34, 3, kAUC=8.7\tPWM= ACAAAAGGTTC\t13.04, 6, pAUC=5.7\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=12 ...\n",
      "\n",
      "k=12, exact kmers=37\n",
      "k=12+1, gapped kmer=0, base kmer=0\n",
      "k=12+2, gapped kmer=0, base kmer=0\n",
      "k=12+3, gapped kmer=0, base kmer=0\n",
      "k=12+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 12 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 37\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.             \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "CACGGCATCGGA/TCCGATGCCGTG\t4\t3\t0\t-1.8    \t6\t4.6\t10.0\t45.8\t6\n",
      "ACAAATTGCAAA/TTTGCAATTTGT\t3\t3\t0\t-1.8    \t0\t3.0\t7.0\t21.0\t3\n",
      "ATCGGACAAAAG/CTTTTGTCCGAT\t4\t3\t0\t-1.8    \t3\t3.0\t7.0\t21.0\t5\n",
      "GCTTCCACGGCA/TGCCGTGGAAGC\t4\t3\t0\t-1.8    \t19\t4.6\t4.0\t18.3\t5\n",
      "ATATTTTGCTTC/GAAGCAAAATAT\t4\t3\t0\t-1.8    \t2\t3.0\t6.0\t18.0\t5\n",
      "\n",
      "Align and cluster 11 k-mers with seed CACGGCATCGGA,   \t#0\n",
      "Align and cluster 10 k-mers with seed ACAAATTGCAAA,   \t#1\n",
      "Align and cluster 8 k-mers with seed ATCGGACAAAAG,   \t#2\n",
      "Align and cluster 9 k-mers with seed GCTTCCACGGCA,   \t#3\n",
      "Align and cluster 11 k-mers with seed ATATTTTGCTTC,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "22ms: Remove motif #1.\n",
      "\n",
      "------------------------- k = 12 ----------------------------\n",
      "\n",
      "#0\tk=12\tKSM= GCTTCCACGGCA \t1.34, 7, kAUC=11.5\tPWM= GCTTCCACGGAA\t10.27, 50, pAUC=50.0\n",
      "#1\tk=12\tKSM= CACGGCATCGGA \t1.34, 3, kAUC=8.7\tPWM= CACGGCATCGGA\t10.48, 12, pAUC=20.6\n",
      "#2\tk=12\tKSM= ATATTTTGCTTC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTC\t14.69, 4, pAUC=9.2\n",
      "#3\tk=12\tKSM= ATCGGACAAAAG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAG\t14.71, 6, pAUC=8.2\n",
      "#4\tk=12\tKSM= ACAAATTGCAAA \t1.22, 3, kAUC=7.8\tPWM= ACAAATTGCAAA\t10.22, 7, pAUC=9.7\n",
      "\n",
      "After merging:\n",
      "#0\tk=12\tKSM= GCTTCCACGGCA \t1.34, 7, kAUC=11.5\tPWM= GCTTCCACGGAA\t10.27, 50, pAUC=50.0\n",
      "#2\tk=12\tKSM= ATATTTTGCTTC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTC\t14.69, 4, pAUC=9.2\n",
      "#3\tk=12\tKSM= ATCGGACAAAAG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAG\t14.71, 6, pAUC=8.2\n",
      "#4\tk=12\tKSM= ACAAATTGCAAA \t1.22, 3, kAUC=7.8\tPWM= ACAAATTGCAAA\t10.22, 7, pAUC=9.7\n",
      "\n",
      "\n",
      "\n",
      "----------------------------------------------------------\n",
      "Running k=13 ...\n",
      "\n",
      "k=13, exact kmers=33\n",
      "k=13+1, gapped kmer=0, base kmer=0\n",
      "k=13+2, gapped kmer=0, base kmer=0\n",
      "k=13+3, gapped kmer=0, base kmer=0\n",
      "k=13+4, gapped kmer=0, base kmer=0\n",
      "\n",
      "------------------------- k = 13 ----------------------------\n",
      "\n",
      "Total number of k-mers: n = 33\n",
      "\n",
      "Top 5 cluster center k-mers:\n",
      "# k-mer/r.c.               \tPosCt\twPosCt\tNegCt\tHgp_lg10\tId\tDensity\tDelta\tGamma\tCluster_size\n",
      "CCACGGCATCGGA/TCCGATGCCGTGG\t4\t3\t0\t-1.8    \t8\t4.6\t11.0\t50.4\t6\n",
      "ATATTTTGCTTCC/GGAAGCAAAATAT\t4\t3\t0\t-1.8    \t1\t3.0\t9.0\t27.0\t5\n",
      "CACAAATTGCAAA/TTTGCAATTTGTG\t3\t3\t0\t-1.8    \t4\t3.0\t8.0\t24.0\t3\n",
      "ATCGGACAAAAGG/CCTTTTGTCCGAT\t4\t3\t0\t-1.8    \t2\t3.0\t7.0\t21.0\t5\n",
      "CGTGGGATATTTT/AAAATATCCCACG\t3\t3\t0\t-1.8    \t12\t3.0\t7.0\t21.0\t3\n",
      "ACGGCATCGGACA/TGTCCGATGCCGT\t4\t3\t0\t-1.8\n",
      "\n",
      "Align and cluster 9 k-mers with seed CCACGGCATCGGA,   \t#0\n",
      "Align and cluster 7 k-mers with seed ATATTTTGCTTCC,   \t#1\n",
      "Align and cluster 5 k-mers with seed CACAAATTGCAAA,   \t#2\n",
      "Align and cluster 7 k-mers with seed ATCGGACAAAAGG,   \t#3\n",
      "Align and cluster 4 k-mers with seed CGTGGGATATTTT,   \t#4\n",
      "\n",
      "Merging redundant motifs ...\n",
      "\n",
      "------------------------- k = 13 ----------------------------\n",
      "\n",
      "#0\tk=13\tKSM= CCACGGCATCGGA \t1.34, 3, kAUC=8.7\tPWM= CCACGGCATCGGA\t15.05, 9, pAUC=20.5\n",
      "#1\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTCC\t15.83, 4, pAUC=16.4\n",
      "#2\tk=13\tKSM= ATCGGACAAAAGG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAGG\t18.37, 5, pAUC=6.3\n",
      "#3\tk=13\tKSM= CGTGGGATATTTT \t1.22, 3, kAUC=7.8\tPWM= GTGGGATCCCT\t9.35, 31, pAUC=29.8\n",
      "#4\tk=13\tKSM= CACAAATTGCAAA \t1.22, 3, kAUC=7.8\tPWM= CACAAATTGCAAA\t10.40, 7, pAUC=8.7\n",
      "\n",
      "After merging:\n",
      "#0\tk=13\tKSM= CCACGGCATCGGA \t1.34, 3, kAUC=8.7\tPWM= CCACGGCATCGGA\t15.05, 9, pAUC=20.5\n",
      "#1\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTCC\t15.83, 4, pAUC=16.4\n",
      "#2\tk=13\tKSM= ATCGGACAAAAGG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAGG\t18.37, 5, pAUC=6.3\n",
      "#3\tk=13\tKSM= CGTGGGATATTTT \t1.22, 3, kAUC=7.8\tPWM= GTGGGATCCCT\t9.35, 31, pAUC=29.8\n",
      "#4\tk=13\tKSM= CACAAATTGCAAA \t1.22, 3, kAUC=7.8\tPWM= CACAAATTGCAAA\t10.40, 7, pAUC=8.7\n",
      "\n",
      "\n",
      "\n",
      "------------------------- out8_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.45, 41, kAUC=53.6\tPWM= TTCCACGG\t8.21, 61, pAUC=54.6\n",
      "#1\tk=9\tKSM= TTCCACGGA \t1.11, 33, kAUC=38.0\tPWM= TTCCACGGA\t8.40, 62, pAUC=55.6\n",
      "#2\tk=10\tKSM= TTCCACGGAA \t1.00, 35, kAUC=35.4\tPWM= TTCCACGGAA\t9.89, 59, pAUC=53.3\n",
      "#3\tk=8\tKSM= ACGGCATC \t1.11, 24, kAUC=25.8\tPWM= ACGGCATC\t8.27, 31, pAUC=19.1\n",
      "#4\tk=11\tKSM= ATTCCACGGAA \t1.00, 18, kAUC=20.7\tPWM= TTCCACGGA\t9.07, 57, pAUC=53.5\n",
      "#5\tk=8\tKSM= TCCTTTCC \t1.11, 14, kAUC=18.0\tPWM= CCCTTTCC\t8.48, 26, pAUC=16.7\n",
      "#6\tk=12\tKSM= GCTTCCACGGCA \t1.34, 7, kAUC=11.5\tPWM= GCTTCCACGGAA\t10.27, 50, pAUC=50.0\n",
      "#7\tk=10\tKSM= CCACTTCCAC \t1.22, 6, kAUC=9.6\tPWM= CCACTTCCAC\t10.28, 22, pAUC=17.0\n",
      "#8\tk=9\tKSM= CAAATTGCA \t1.00, 4, kAUC=9.6\tPWM= CAAATTGCA\t7.92, 15, pAUC=14.6\n",
      "#9\tk=9\tKSM= CCCCTCCCC \t1.34, 6, kAUC=9.6\tPWM= CCCCCCCCC\t8.83, 20, pAUC=12.3\n",
      "#10\tk=13\tKSM= CCACGGCATCGGA \t1.34, 3, kAUC=8.7\tPWM= CCACGGCATCGGA\t15.05, 9, pAUC=20.5\n",
      "#11\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTCC\t15.83, 4, pAUC=16.4\n",
      "#12\tk=12\tKSM= ATATTTTGCTTC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTC\t14.69, 4, pAUC=9.2\n",
      "#13\tk=11\tKSM= ATATTTTGCTT \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTT\t13.11, 5, pAUC=8.8\n",
      "#14\tk=12\tKSM= ATCGGACAAAAG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAG\t14.71, 6, pAUC=8.2\n",
      "#15\tk=13\tKSM= ATCGGACAAAAGG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAGG\t18.37, 5, pAUC=6.3\n",
      "#16\tk=11\tKSM= ACAAAAGGTTC \t1.34, 3, kAUC=8.7\tPWM= ACAAAAGGTTC\t13.04, 6, pAUC=5.7\n",
      "#17\tk=10\tKSM= ACAAAAGGTT \t1.34, 3, kAUC=8.7\tPWM= ACAAAAGGTT\t15.89, 3, pAUC=3.9\n",
      "#18\tk=13\tKSM= CGTGGGATATTTT \t1.22, 3, kAUC=7.8\tPWM= GTGGGATCCCT\t9.35, 31, pAUC=29.8\n",
      "#19\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=7.8\tPWM= GGTGCTGCTC\t10.75, 11, pAUC=12.3\n",
      "#20\tk=12\tKSM= ACAAATTGCAAA \t1.22, 3, kAUC=7.8\tPWM= ACAAATTGCAAA\t10.22, 7, pAUC=9.7\n",
      "#21\tk=13\tKSM= CACAAATTGCAAA \t1.22, 3, kAUC=7.8\tPWM= CACAAATTGCAAA\t10.40, 7, pAUC=8.7\n",
      "#22\tk=9\tKSM= CATACCACC \t1.34, 4, kAUC=7.8\tPWM= CATACCACC\t10.22, 7, pAUC=7.1\n",
      "\n",
      "35ms: Remove motif #1.\n",
      "45ms: Remove motif #2.\n",
      "55ms: Remove motif #4.\n",
      "56ms: PWM GCTTCCACGGAA has been removed.\n",
      "67ms: Remove motif #10.\n",
      "76ms: Remove motif #18.\n",
      "77ms: PWM ACAAATTGCAAA has been removed.\n",
      "79ms: PWM CACAAATTGCAAA has been removed.\n",
      "89ms: Remove motif #12.\n",
      "99ms: Remove motif #13.\n",
      "100ms: PWM ATCGGACAAAAGG has been removed.\n",
      "127ms: Remove motif #16.\n",
      "128ms: PWM ACAAAAGGTT has been removed.\n",
      "\n",
      "------------------------- out8_2, k = ALL (before merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.45, 41, kAUC=53.6\tPWM= TTCCACGG\t8.21, 61, pAUC=54.6\n",
      "#1\tk=9\tKSM= TTCCACGGA \t1.11, 33, kAUC=38.0\tPWM= TTCCACGGA\t8.40, 62, pAUC=55.6\n",
      "#2\tk=10\tKSM= TTCCACGGAA \t1.00, 35, kAUC=35.4\tPWM= TTCCACGGAA\t9.89, 59, pAUC=53.3\n",
      "#3\tk=8\tKSM= ACGGCATC \t1.11, 24, kAUC=25.8\tPWM= ACGGCATC\t8.27, 31, pAUC=19.1\n",
      "#4\tk=11\tKSM= ATTCCACGGAA \t1.00, 18, kAUC=20.7\tPWM= TTCCACGGA\t9.07, 57, pAUC=53.5\n",
      "#5\tk=8\tKSM= TCCTTTCC \t1.11, 14, kAUC=18.0\tPWM= CCCTTTCC\t8.48, 26, pAUC=16.7\n",
      "#6\tk=12\tKSM= GCTTCCACGGCA \t1.34, 7, kAUC=11.5\tPWM= GCTTCCACGGAA\t10.27, 50, pAUC=50.0\n",
      "#7\tk=10\tKSM= CCACTTCCAC \t1.22, 6, kAUC=9.6\tPWM= CCACTTCCAC\t10.28, 22, pAUC=17.0\n",
      "#8\tk=9\tKSM= CAAATTGCA \t1.00, 4, kAUC=9.6\tPWM= CAAATTGCA\t7.92, 15, pAUC=14.6\n",
      "#9\tk=9\tKSM= CCCCTCCCC \t1.34, 6, kAUC=9.6\tPWM= CCCCCCCCC\t8.83, 20, pAUC=12.3\n",
      "#10\tk=13\tKSM= CCACGGCATCGGA \t1.34, 3, kAUC=8.7\tPWM= CCACGGCATCGGA\t15.05, 9, pAUC=20.5\n",
      "#11\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTCC\t15.83, 4, pAUC=16.4\n",
      "#12\tk=12\tKSM= ATATTTTGCTTC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTC\t14.69, 4, pAUC=9.2\n",
      "#13\tk=11\tKSM= ATATTTTGCTT \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTT\t13.11, 5, pAUC=8.8\n",
      "#14\tk=12\tKSM= ATCGGACAAAAG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAG\t14.71, 6, pAUC=8.2\n",
      "#15\tk=13\tKSM= ATCGGACAAAAGG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAGG\t18.37, 5, pAUC=6.3\n",
      "#16\tk=11\tKSM= ACAAAAGGTTC \t1.34, 3, kAUC=8.7\tPWM= ACAAAAGGTTC\t13.04, 6, pAUC=5.7\n",
      "#17\tk=10\tKSM= ACAAAAGGTT \t1.34, 3, kAUC=8.7\tPWM= ACAAAAGGTT\t15.89, 3, pAUC=3.9\n",
      "#18\tk=13\tKSM= CGTGGGATATTTT \t1.22, 3, kAUC=7.8\tPWM= GTGGGATCCCT\t9.35, 31, pAUC=29.8\n",
      "#19\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=7.8\tPWM= GGTGCTGCTC\t10.75, 11, pAUC=12.3\n",
      "#20\tk=12\tKSM= ACAAATTGCAAA \t1.22, 3, kAUC=7.8\tPWM= ACAAATTGCAAA\t10.22, 7, pAUC=9.7\n",
      "#21\tk=13\tKSM= CACAAATTGCAAA \t1.22, 3, kAUC=7.8\tPWM= CACAAATTGCAAA\t10.40, 7, pAUC=8.7\n",
      "#22\tk=9\tKSM= CATACCACC \t1.34, 4, kAUC=7.8\tPWM= CATACCACC\t10.22, 7, pAUC=7.1\n",
      "\n",
      "------------------------- out8_2, k = ALL (after merging) -------------------------\n",
      "#0\tk=8\tKSM= TTCCACGG \t1.45, 41, kAUC=53.6\tPWM= TTCCACGG\t8.21, 61, pAUC=54.6\n",
      "#3\tk=8\tKSM= ACGGCATC \t1.11, 24, kAUC=25.8\tPWM= ACGGCATC\t8.27, 31, pAUC=19.1\n",
      "#5\tk=8\tKSM= TCCTTTCC \t1.11, 14, kAUC=18.0\tPWM= CCCTTTCC\t8.48, 26, pAUC=16.7\n",
      "#7\tk=10\tKSM= CCACTTCCAC \t1.22, 6, kAUC=9.6\tPWM= CCACTTCCAC\t10.28, 22, pAUC=17.0\n",
      "#8\tk=9\tKSM= CAAATTGCA \t1.00, 4, kAUC=9.6\tPWM= CAAATTGCA\t7.92, 15, pAUC=14.6\n",
      "#9\tk=9\tKSM= CCCCTCCCC \t1.34, 6, kAUC=9.6\tPWM= CCCCCCCCC\t8.83, 20, pAUC=12.3\n",
      "#11\tk=13\tKSM= ATATTTTGCTTCC \t1.00, 3, kAUC=8.7\tPWM= ATATTTTGCTTCC\t15.83, 4, pAUC=16.4\n",
      "#14\tk=12\tKSM= ATCGGACAAAAG \t1.34, 3, kAUC=8.7\tPWM= ATCGGACAAAAG\t14.71, 6, pAUC=8.2\n",
      "#19\tk=10\tKSM= GGTGCTGCTC \t1.34, 3, kAUC=7.8\tPWM= GGTGCTGCTC\t10.75, 11, pAUC=12.3\n",
      "#22\tk=9\tKSM= CATACCACC \t1.34, 4, kAUC=7.8\tPWM= CATACCACC\t10.22, 7, pAUC=7.1\n",
      "\n",
      "151ms: Finalizing 10 motifs ...\n",
      "\n",
      "\n",
      "Compute motif distance distribution ...\n",
      "\n",
      "------------------------- out8_2 final motifs ----------------------------\n",
      "\n",
      "\n",
      "out8_2 motif #0\n",
      "\n",
      "KSM top k-mer: TTCCACGG/CCGTGGAA, total 46 k-mers\n",
      "KSM threshold: 1.45, \t\thit=41+/0-, kAUC=53.6\n",
      "\n",
      "                |\n",
      "TTCCACGG\n",
      " C      \n",
      "        \n",
      "        \n",
      "PWM threshold: 8.21/15.03, \thit=61+/4-, pAUC=54.6\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #1\n",
      "\n",
      "KSM top k-mer: ACGGCATC/GATGCCGT, total 15 k-mers\n",
      "KSM threshold: 1.11, \t\thit=24+/0-, kAUC=25.8\n",
      "\n",
      "             |\n",
      "ACGGCATC\n",
      "C     A \n",
      "        \n",
      "        \n",
      "PWM threshold: 8.27/12.45, \thit=31+/9-, pAUC=19.1\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #2\n",
      "\n",
      "KSM top k-mer: TCCTTTCC/GGAAAGGA, total 8 k-mers\n",
      "KSM threshold: 1.11, \t\thit=14+/0-, kAUC=18.0\n",
      "\n",
      "|\n",
      "CCCTTTCC\n",
      "T       \n",
      "        \n",
      "        \n",
      "PWM threshold: 8.48/13.67, \thit=26+/8-, pAUC=16.7\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #3\n",
      "\n",
      "KSM top k-mer: CCACTTCCAC/GTGGAAGTGG, total 4 k-mers\n",
      "KSM threshold: 1.22, \t\thit=6+/0-, kAUC=9.6\n",
      "\n",
      "                    |\n",
      "CCACTTCCAC\n",
      "   GGG  CG\n",
      "          \n",
      "          \n",
      "PWM threshold: 10.28/15.32, \thit=22+/3-, pAUC=17.0\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #4\n",
      "\n",
      "KSM top k-mer: CAAATTGCA/TGCAATTTG, total 6 k-mers\n",
      "KSM threshold: 1.00, \t\thit=4+/0-, kAUC=9.6\n",
      "\n",
      "CAAATTGCA\n",
      "  C GC\n",
      "PWM threshold: 7.92/13.13, \thit=15+/3-, pAUC=14.6\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #5\n",
      "\n",
      "KSM top k-mer: CCCCTCCCC/GGGGAGGGG, total 2 k-mers\n",
      "KSM threshold: 1.34, \t\thit=6+/0-, kAUC=9.6\n",
      "\n",
      "       |\n",
      "CCCCCCCCC\n",
      " G  T    \n",
      "         \n",
      "         \n",
      "PWM threshold: 8.83/16.16, \thit=20+/8-, pAUC=12.3\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #6\n",
      "\n",
      "KSM top k-mer: ATATTTTGCTTCC/GGAAGCAAAATAT, total 7 k-mers\n",
      "KSM threshold: 1.00, \t\thit=3+/0-, kAUC=8.7\n",
      "\n",
      "                         |\n",
      "ATATTTTGCTTCC\n",
      "             \n",
      "             \n",
      "             \n",
      "PWM threshold: 15.83/20.61, \thit=4+/0-, pAUC=16.4\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #7\n",
      "\n",
      "KSM top k-mer: ATCGGACAAAAG/CTTTTGTCCGAT, total 7 k-mers\n",
      "KSM threshold: 1.34, \t\thit=3+/0-, kAUC=8.7\n",
      "\n",
      "       |\n",
      "ATCGGACAAAAG\n",
      " C    G  C  \n",
      "            \n",
      "            \n",
      "PWM threshold: 14.71/19.04, \thit=6+/0-, pAUC=8.2\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #8\n",
      "\n",
      "KSM top k-mer: GGTGCTGCTC/GAGCAGCACC, total 1 k-mers\n",
      "KSM threshold: 1.34, \t\thit=3+/0-, kAUC=7.8\n",
      "\n",
      "GGTGCTGCTC\n",
      "PWM threshold: 10.75/17.78, \thit=11+/0-, pAUC=12.3\n",
      "--------------------------------------------------------------\n",
      "\n",
      "out8_2 motif #9\n",
      "\n",
      "KSM top k-mer: CATACCACC/GGTGGTATG, total 1 k-mers\n",
      "KSM threshold: 1.34, \t\thit=4+/0-, kAUC=7.8\n",
      "\n",
      "         |\n",
      "CATACCACC\n",
      "         \n",
      "         \n",
      "         \n",
      "PWM threshold: 10.22/14.96, \thit=7+/0-, pAUC=7.1\n",
      "--------------------------------------------------------------\n",
      "\n",
      "\n",
      "\n",
      "Finished! GEM analysis results are printed to:\n",
      "/home/boycem4/chip_seq_project/all/out8/out8.GEM_events.txt\n",
      "/home/boycem4/chip_seq_project/all/out8/out8.results.htm\n",
      "/home/boycem4/chip_seq_project/all/out8/out8_outputs (folder with all other files)\n",
      "\n",
      "----------------------------------\n",
      "\n",
      "End time: 2022/01/11 19:18:29\n",
      "\n",
      "Total running time: 1.8m\n",
      "\n"
     ]
    }
   ],
   "source": [
    "#outbed whole genome with quality control trimming at 40 and PCR removed\n",
    "!java -Xmx4G -jar /usr/local/gem/gem.jar \\\n",
    "    --t 1 \\\n",
    "    --expt chip_q30_rmdup.bam \\\n",
    "    --f BAM --genome . \\\n",
    "    --g chrom_sizes.txt \\\n",
    "    --d Read_Distribution_default.txt \\\n",
    "    --k_min 8 --k_max 13 \\\n",
    "    --ctrl ctrl_q30_rmdup.bam \\\n",
    "    --out out8 \\\n",
    "    --outBED "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
