{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Notebook with system commands to process FastQ files: mapping with Bowtie2, sorting and indexing with samtools, conversion to BigWigs with bamCoverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-r--r-- 1 boycem4 GEU33025 53M Nov 25  2020 ctrl_R1.fastq.gz\n",
      "-rw-r--r-- 1 boycem4 GEU33025 53M Nov 30  2020 ctrl_R2.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "# list all files ending in q.gz\n",
    "!ls -lh *q.gz"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# settings for subsequent commands\n",
    "\n",
    "# select the sample to be processed:\n",
    "sample = 'ctrl'\n",
    "\n",
    "# prefix for output files can be the same as sample name\n",
    "out_prefix = sample\n",
    "\n",
    "# reference genome\n",
    "reference = 'C_glabrata.fa'\n",
    "\n",
    "# name for bowtie2 index files\n",
    "bowtie_index = 'C_glabrata'\n",
    "\n",
    "# number of threads to use in parallel\n",
    "threads = 1\n",
    "\n",
    "# execute stepwise or as a pipeline\n",
    "#processing = 'stepwise'\n",
    "processing = 'pipeline'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "file C_glabrata.1.bt2 does not exist\n",
      "job: bowtie2-build C_glabrata.fa C_glabrata\n",
      "Settings:\n",
      "  Output files: \"C_glabrata.*.bt2\"\n",
      "  Line rate: 6 (line is 64 bytes)\n",
      "  Lines per side: 1 (side is 64 bytes)\n",
      "  Offset rate: 4 (one in 16)\n",
      "  FTable chars: 10\n",
      "  Strings: unpacked\n",
      "  Max bucket size: default\n",
      "  Max bucket size, sqrt multiplier: default\n",
      "  Max bucket size, len divisor: 4\n",
      "  Difference-cover sample period: 1024\n",
      "  Endianness: little\n",
      "  Actual local endianness: little\n",
      "  Sanity checking: disabled\n",
      "  Assertions: disabled\n",
      "  Random seed: 0\n",
      "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
      "Input files DNA, FASTA:\n",
      "  C_glabrata.fa\n",
      "Building a SMALL index\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "bmax according to bmaxDivN setting: 3084501\n",
      "Using parameters --bmax 2313376 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 2313376 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:01\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.2338e+07 (target: 2313375)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 12338004 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:01\n",
      "Returning block of 12338005 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 3783933\n",
      "fchr[G]: 6169684\n",
      "fchr[T]: 8548676\n",
      "fchr[$]: 12338004\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 8307483 bytes to primary EBWT file: C_glabrata.1.bt2\n",
      "Wrote 3084508 bytes to secondary EBWT file: C_glabrata.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 12338004\n",
      "    bwtLen: 12338005\n",
      "    sz: 3084501\n",
      "    bwtSz: 3084502\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 771126\n",
      "    offsSz: 3084504\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 64261\n",
      "    numLines: 64261\n",
      "    ebwtTotLen: 4112704\n",
      "    ebwtTotSz: 4112704\n",
      "    color: 0\n",
      "    reverse: 0\n",
      "Total time for call to driver() for forward index: 00:00:03\n",
      "Reading reference sizes\n",
      "  Time reading reference sizes: 00:00:00\n",
      "Calculating joined length\n",
      "Writing header\n",
      "Reserving space for joined string\n",
      "Joining reference sequences\n",
      "  Time to join reference sequences: 00:00:00\n",
      "  Time to reverse reference sequence: 00:00:00\n",
      "bmax according to bmaxDivN setting: 3084501\n",
      "Using parameters --bmax 2313376 --dcv 1024\n",
      "  Doing ahead-of-time memory usage test\n",
      "  Passed!  Constructing with these parameters: --bmax 2313376 --dcv 1024\n",
      "Constructing suffix-array element generator\n",
      "Building DifferenceCoverSample\n",
      "  Building sPrime\n",
      "  Building sPrimeOrder\n",
      "  V-Sorting samples\n",
      "  V-Sorting samples time: 00:00:00\n",
      "  Allocating rank array\n",
      "  Ranking v-sort output\n",
      "  Ranking v-sort output time: 00:00:00\n",
      "  Invoking Larsson-Sadakane on ranks\n",
      "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
      "  Sanity-checking and returning\n",
      "Building samples\n",
      "Reserving space for 12 sample suffixes\n",
      "Generating random suffixes\n",
      "QSorting 12 sample offsets, eliminating duplicates\n",
      "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
      "Multikey QSorting 12 samples\n",
      "  (Using difference cover)\n",
      "  Multikey QSorting samples time: 00:00:00\n",
      "Calculating bucket sizes\n",
      "Splitting and merging\n",
      "  Splitting and merging time: 00:00:00\n",
      "Avg bucket size: 1.2338e+07 (target: 2313375)\n",
      "Converting suffix-array elements to index image\n",
      "Allocating ftab, absorbFtab\n",
      "Entering Ebwt loop\n",
      "Getting block 1 of 1\n",
      "  No samples; assembling all-inclusive block\n",
      "  Sorting block of length 12338004 for bucket 1\n",
      "  (Using difference cover)\n",
      "  Sorting block time: 00:00:02\n",
      "Returning block of 12338005 for bucket 1\n",
      "Exited Ebwt loop\n",
      "fchr[A]: 0\n",
      "fchr[C]: 3783933\n",
      "fchr[G]: 6169684\n",
      "fchr[T]: 8548676\n",
      "fchr[$]: 12338004\n",
      "Exiting Ebwt::buildToDisk()\n",
      "Returning from initFromVector\n",
      "Wrote 8307483 bytes to primary EBWT file: C_glabrata.rev.1.bt2\n",
      "Wrote 3084508 bytes to secondary EBWT file: C_glabrata.rev.2.bt2\n",
      "Re-opening _in1 and _in2 as input streams\n",
      "Returning from Ebwt constructor\n",
      "Headers:\n",
      "    len: 12338004\n",
      "    bwtLen: 12338005\n",
      "    sz: 3084501\n",
      "    bwtSz: 3084502\n",
      "    lineRate: 6\n",
      "    offRate: 4\n",
      "    offMask: 0xfffffff0\n",
      "    ftabChars: 10\n",
      "    eftabLen: 20\n",
      "    eftabSz: 80\n",
      "    ftabLen: 1048577\n",
      "    ftabSz: 4194308\n",
      "    offsLen: 771126\n",
      "    offsSz: 3084504\n",
      "    lineSz: 64\n",
      "    sideSz: 64\n",
      "    sideBwtSz: 48\n",
      "    sideBwtLen: 192\n",
      "    numSides: 64261\n",
      "    numLines: 64261\n",
      "    ebwtTotLen: 4112704\n",
      "    ebwtTotSz: 4112704\n",
      "    color: 0\n",
      "    reverse: 1\n",
      "Total time for backward call to driver() for mirror index: 00:00:02\n",
      "\n",
      "real\t0m4.922s\n",
      "user\t0m4.688s\n",
      "sys\t0m0.145s\n"
     ]
    }
   ],
   "source": [
    "import os.path\n",
    "\n",
    "# make sure the mapping index files are in place\n",
    "for i in range(1,5) :\n",
    "    bowtie_index_file = f'{bowtie_index}.{i}.bt2'\n",
    "\n",
    "    if not os.path.isfile(bowtie_index_file) :\n",
    "        \n",
    "        print(f'file {bowtie_index_file} does not exist')\n",
    "        \n",
    "        # try to generate the index\n",
    "        job = f'bowtie2-build {reference} {bowtie_index}'\n",
    "        print(f'job: {job}')\n",
    "        !time {job}\n",
    "        \n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define the input file(s)\n",
    "# (for paired-end reads we have two input files):\n",
    "infile1 = f'{sample}_R1.fastq.gz'\n",
    "infile2 = f'{sample}_R2.fastq.gz'\n",
    "\n",
    "# one could also use trimmed files if available:\n",
    "#infile1 = f'{sample}_R1_val_1.fq.gz'\n",
    "#infile2 = f'{sample}_R2_val_2.fq.gz'\n",
    "\n",
    "# make sure input files are in place\n",
    "for file in (infile1, infile2) :\n",
    "    if not os.path.isfile(file) :\n",
    "        raise Exception(f'file {file} does not exist')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "job: bowtie2 -x C_glabrata -1 ctrl_R1.fastq.gz -2 ctrl_R2.fastq.gz --mm -p 1 | samtools view -b | samtools sort -@ 1 > ctrl.bam\n",
      "1071087 reads; of these:\n",
      "  1071087 (100.00%) were paired; of these:\n",
      "    304520 (28.43%) aligned concordantly 0 times\n",
      "    669889 (62.54%) aligned concordantly exactly 1 time\n",
      "    96678 (9.03%) aligned concordantly >1 times\n",
      "    ----\n",
      "    304520 pairs aligned concordantly 0 times; of these:\n",
      "      50408 (16.55%) aligned discordantly 1 time\n",
      "    ----\n",
      "    254112 pairs aligned 0 times concordantly or discordantly; of these:\n",
      "      508224 mates make up the pairs; of these:\n",
      "        468853 (92.25%) aligned 0 times\n",
      "        23234 (4.57%) aligned exactly 1 time\n",
      "        16137 (3.18%) aligned >1 times\n",
      "78.11% overall alignment rate\n",
      "\n",
      "real\t2m3.030s\n",
      "user\t2m48.292s\n",
      "sys\t0m2.367s\n"
     ]
    }
   ],
   "source": [
    "if processing == 'stepwise' :\n",
    "    \n",
    "    # run bowtie2 mapper and save output in SAM format\n",
    "    job = f'bowtie2 -x {bowtie_index} -1 {infile1} -2 {infile2} --mm -p {threads} > {out_prefix}.sam'\n",
    "    print(f'job: {job}')\n",
    "    !time {job}\n",
    "    \n",
    "    # convert SAM into BAM format, so that it can be sorted\n",
    "    job = f'samtools view -b {out_prefix}.sam > {out_prefix}.bam'\n",
    "    print(f'job: {job}')\n",
    "    !time {job}\n",
    "    \n",
    "    # sort BAM file (requires BAM file as input)\n",
    "    job = f'samtools sort -@ {threads} {out_prefix}.bam > {out_prefix}.sorted.bam'\n",
    "    print(f'job: {job}')\n",
    "    !time {job}\n",
    "    \n",
    "    # replace unsorted BAM file with sorted one\n",
    "    job = f'mv {out_prefix}.sorted.bam {out_prefix}.bam'\n",
    "    print(f'job: {job}')\n",
    "    !time {job}\n",
    "    \n",
    "else :\n",
    "    \n",
    "    # run bowtie2 mapper and save compressed and sorted output in {out_prefix}.bam in one go\n",
    "    job = f'bowtie2 -x {bowtie_index} -1 {infile1} -2 {infile2} --mm -p {threads} | samtools view -b | samtools sort -@ {threads} > {out_prefix}.bam'\n",
    "    print(f'job: {job}')\n",
    "    !time {job}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access '*.sam': No such file or directory\n",
      "-rw-r--r-- 1 boycem4 GEU3302521 3.7M Dec  1 19:46  chipA.bam\n",
      "-rw-r--r-- 1 boycem4 GEU3302521  85M Dec  3 11:25  ctrl.bam\n"
     ]
    }
   ],
   "source": [
    "# list all the SAM and BAM files in the directory\n",
    "!ls -lh *.bam *.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "job: samtools index ctrl.bam\n",
      "\n",
      "real\t0m1.259s\n",
      "user\t0m1.244s\n",
      "sys\t0m0.012s\n"
     ]
    }
   ],
   "source": [
    "# create an index for the BAM file to allow direct access at certain positions\n",
    "job = f'samtools index {out_prefix}.bam'\n",
    "print(f'job: {job}')\n",
    "!time {job}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "job: samtools view -b ctrl.bam chrA > ctrlA.bam\n",
      "\n",
      "real\t0m0.526s\n",
      "user\t0m0.491s\n",
      "sys\t0m0.012s\n",
      "job: samtools index ctrlA.bam\n",
      "\n",
      "real\t0m0.089s\n",
      "user\t0m0.082s\n",
      "sys\t0m0.004s\n"
     ]
    }
   ],
   "source": [
    "# create a BAM file from reads mapped to chrA only\n",
    "job = f'samtools view -b {out_prefix}.bam chrA > {out_prefix}A.bam'\n",
    "print(f'job: {job}')\n",
    "!time {job}\n",
    "\n",
    "# index this BAM file\n",
    "job = f'samtools index {out_prefix}A.bam'\n",
    "print(f'job: {job}')\n",
    "!time {job}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "job: /usr/local/deeptools3/bin/bamCoverage -p 1 -b ctrl.bam -o ctrl.bw\n",
      "bamFilesList: ['ctrl.bam']\n",
      "binLength: 50\n",
      "numberOfSamples: None\n",
      "blackListFileName: None\n",
      "skipZeroOverZero: False\n",
      "bed_and_bin: False\n",
      "genomeChunkSize: None\n",
      "defaultFragmentLength: read length\n",
      "numberOfProcessors: 1\n",
      "verbose: False\n",
      "region: None\n",
      "bedFile: None\n",
      "minMappingQuality: None\n",
      "ignoreDuplicates: False\n",
      "chrsToSkip: []\n",
      "stepSize: 50\n",
      "center_read: False\n",
      "samFlag_include: None\n",
      "samFlag_exclude: None\n",
      "minFragmentLength: 0\n",
      "maxFragmentLength: 0\n",
      "zerosToNans: False\n",
      "smoothLength: None\n",
      "save_data: False\n",
      "out_file_for_raw_data: None\n",
      "maxPairedFragmentLength: 1000\n",
      "\n",
      "real\t0m12.982s\n",
      "user\t0m11.926s\n",
      "sys\t0m0.140s\n"
     ]
    }
   ],
   "source": [
    "# generate BigWig files from BAM files\n",
    "job = f'/usr/local/deeptools3/bin/bamCoverage -p {threads} -b {out_prefix}.bam -o {out_prefix}.bw'\n",
    "print(f'job: {job}')\n",
    "!time {job}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-r--r-- 1 boycem4 GEU3302521 712K Dec  3 11:26 ctrl.bw\n"
     ]
    }
   ],
   "source": [
    "# list all BigWig files:\n",
    "!ls -lh *.bw"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
